---
title: "HBB-TF Regression Analysis"
author: "Yifan Zhou"
output:
  html_document: 
    toc: true
---

```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.width=6, fig.height=4, fig.path='Figs/',
                      echo=TRUE, warning=FALSE, message=FALSE, comment = NA)
```

## Load Data

We choose whole blood as the tissue of interest for preliminary analysis.

Load tissue-specific (whole blood) gene expression tpm data extracted from GTEx v7 `GTEx_Analysis_2016-01-15_v7_RNASeQCv1.1.8_gene_tpm.gct`, and do normalization:

```{r}
load('~/Downloads/Research/GTEx/blood_gene_exp_norm.Rdata')
```

Load data from 'tissue-specific circuits' paper (`blood_adult.txt` from [394 individual networks] http://regulatorycircuits.org/download.html):
```{r}
blood_TF_gene <- read.delim(file = '~/Downloads/Research/GTEx/blood_adult_TF_gene.txt',
                            sep = '\t', header = FALSE, stringsAsFactors = FALSE) 
```

## _HBB_ and its Associated TFs

We choose _HBB_, a gene that encodes haemoglobin subunit beta as a target gene of interest to analyze the association of _HBB_ expression level with the expression levels of putative TFs linked to _HBB_:  
```{r}
HBB <- blood_TF_gene[which(blood_TF_gene[,2]=='HBB'),]
HBB <- HBB[order(HBB[,3],decreasing = TRUE),]
HBB_TF_names<- HBB[,1]
```

Putative TF list for gene _HBB_:
```{r}
as.character(HBB_TF_names)
```

Search for these 16 TFs in the GTEx gene expression list:
```{r}
for (i in HBB_TF_names){
    print(blood_gene_exp_norm$Description[startsWith(blood_gene_exp_norm$Description, as.character(i))] )
}
```

3 TFs, _CACD_, _RAR_, and _HF1H3B_, cannot be found in GTEx.

Extract expression data of the rest 13 TFs that are available in GTEx:
```{r}
HBB_TF_names <- HBB_TF_names[c(1,3,4,5,8:16)]
n = 13
HBB_TF_exp <- data.frame(matrix(nrow = 407, ncol = n))
for (i in c(1:n)){
  HBB_TF_exp[,i] <- t(blood_gene_exp_norm[which(blood_gene_exp_norm$Description
                                         ==as.character(HBB_TF_names[i])),-1])
}

HBB_exp <- blood_gene_exp_norm[which(blood_gene_exp_norm$Description=='HBB'),-1]
HBB_TF_exp <- cbind(HBB_TF_exp,t(HBB_exp))
colnames(HBB_TF_exp) <- c(HBB_TF_names,'HBB')
row.names(HBB_TF_exp) <- colnames(blood_gene_exp_norm)[-1]
```


## Linear Regression

Multiple linear regression on all 13 TFs:
```{r}
fit_all_HBB <- lm(HBB ~ ., data = HBB_TF_exp)
summary(fit_all_HBB)
```

**Forward Selection**
```{r echo=FALSE}
df <- HBB_TF_exp
response_str <- "HBB ~ "
end <- length(colnames(df))
R <- colnames(df)[-end] # Initial set of predictors
pval <- rep(NA,length(R)); names(pval) <- R
R_sq <- rep(NA,length(R)); names(R_sq) <- R
for(j in R){
  fit <- lm(as.formula(paste0(response_str,j)), data=df)
  fitsum <- summary(fit)
  pval[j] <- fitsum$coefficients[j,4]
  R_sq[j] <- fitsum$r.squared
}
min_pval <- pval[which(pval == min(pval))]
r_sq <- R_sq[which(pval == min(pval))]
R[which(pval == min(pval))]
```

After the first predictor _GMEB2_ is selected,
add rest of the predictors one at a time until none of their p-values are significant (p value > 0.05).
```{r echo=TRUE}
stats <- data.frame(GMEB2 = c(min_pval,r_sq),row.names = c("p-val","R-squared"))
fit <- lm(as.formula(paste0(response_str,names(min_pval))), data=df)
R <- R[-which(R == names(min_pval))] # The rest of the predictors
nonstop = TRUE
while(nonstop){
  pval <- rep(NA,length(R)); names(pval) <- R
  R_sq <- rep(NA,length(R)); names(R_sq) <- R
  for(j in R){
    fit_loop <- update(fit, as.formula(paste0(".~. +",j)))
    fitsum <- summary(fit_loop)
    pval[j] <- fitsum$coefficients[j,4]
    R_sq[j] <- fitsum$r.squared
  }
  min_pval <- pval[which(pval == min(pval))]
  r_sq <- R_sq[which(pval == min(pval))]
  if(min_pval < 0.05){
    stats[,names(min_pval)] <- c(min_pval,r_sq)
    fit <- update(fit, as.formula(paste0(".~. +",names(min_pval))))
    R <- R[-which(R == names(min_pval))]
  }
  else{
    nonstop = FALSE
    forward_model <- fit
  }
}
rm(list = 'fit_loop')
```

The p-value and $R^2$ statistic for each of the significant predictors added at each round of selection:
```{r}
stats
```

Summary of the selected model:
```{r}
summary(forward_model)
```

```{r}
plot(as.matrix(stats)[2,], pch=16, xlab="Number of Predictors",ylab = "R_2")
```

Check the normality assumption:
```{r}
par(mfrow=c(1,2))
hist(forward_model$residuals)
qqnorm(forward_model$residuals,pch=16) 
qqline(forward_model$residuals,col='red')
```

**Bayesian Information Criterion (BIC)**

```{r}
library(leaps)
RSS <- summary(regsubsets(HBB ~., HBB_TF_exp)) 
n <- dim(HBB_TF_exp)[1]
p <- length(RSS$rss)+1
BIC <- n*log(RSS$rss/n) + log(n)*c(2:p) 
plot(BIC,xlab='Number of Predictors', pch=16)
```

BIC is maximized when the following 6 predictors are used:
```{r}
BIC_predictors <- RSS$which[6,RSS$which[6,]]
BIC_predictors <- names(BIC_predictors)[-1]
BIC_predictors
```

BIC selected model:
```{r}
BIC_model <- lm(HBB  ~ KLF4 + SP1 + GATA6 + NFIC + HNF4A + GMEB2, data = HBB_TF_exp)
summary(BIC_model)
```


**Normality assumption**

```{r}
par(mfrow=c(1,2))
hist(BIC_model$residuals)
qqnorm(BIC_model$residuals,pch=16) 
qqline(BIC_model$residuals,col='red')
```


## Negative Controls

```{r eval=FALSE, include=FALSE}
p = 6
r = 10000
R_sq_HBB = rep(0,r)
HBB_TF_control <- data.frame(matrix(nrow = 407, ncol = p))
HBB_TF_control <- cbind(HBB_TF_control,t(HBB_exp))
colnames(HBB_TF_control)[p+1] <- 'HBB'

rand_HBB = matrix(nrow = r, ncol = p)

for (k in c(1:r)){
  rand_HBB[k,] <- sample(c(1:56202),p,replace = FALSE)
  rand_HBB_name <- blood_gene_exp_norm$Description[rand_HBB[k,]]
  temp <- rand_HBB_name[duplicated(rand_HBB_name)]

  while (length(temp)>0){
    rand_HBB[k,] <- sample(c(1:56202),p,replace = FALSE)
    rand_HBB_name <- blood_gene_exp_norm$Description[rand_HBB[k,]]
    temp <- rand_HBB_name[duplicated(rand_HBB_name)]
  }

  colnames(HBB_TF_control)[1:p] <- rand_HBB_name

  for (i in c(1:p)){
    HBB_TF_control[,i] <- t(blood_gene_exp_norm[rand_HBB[k,i],-1])
  }

  fit_control <- summary(lm(HBB ~. , data = HBB_TF_control))
  R_sq_HBB[k] = fit_control$r.squared
}
```

$R^2$s from the control models:
```{r}
hist(R_sq_HBB, main = 'Distribution of R_sq of Control Models')
summary(R_sq_HBB)
```




