---
title: "Gene-TF Regression Analysis"
author: "Yifan Zhou"
output:
  html_document:
    toc: false
---

```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.width=6, fig.height=4, fig.path='Figs/',
                      echo=TRUE, warning=FALSE, message=FALSE, comment = NA)
```

We choose whole blood as the tissue of interest for preliminary analysis.

Load tissue-specific (whole blood) gene expression tpm data extraced from GTEx v7 `GTEx_Analysis_2016-01-15_v7_RNASeQCv1.1.8_gene_tpm.gct`:

```{r}
blood_gene_exp <- read.delim(file = 'Data/blood_gene_expression.txt',
                             sep = '\t',header = TRUE,stringsAsFactors = FALSE)
```

Load data from 'tissue-specific circuits' paper (`blood_adult.txt` from [394 individual networks] http://regulatorycircuits.org/download.html):
```{r}
blood_TF_gene <- read.delim(file = 'Data/blood_adult_TF_gene.txt',sep = '\t',
                            header = FALSE,stringsAsFactors = TRUE) 
gene_names <-blood_TF_gene[,2]
```

We choose _HBB_, a gene that encodes haemoglobin subunit beta as a target gene of interest to analyze the association of _HBB_ expression level with the expression levels of putative TFs linked to _HBB_:  
```{r}
HBB <- blood_TF_gene[which(gene_names=='HBB'),]
temp <- HBB[,3]
HBB <- HBB[order(temp,decreasing = TRUE),]
HBB_TF_names<- HBB[,1]
```

Putative TF list for gene _HBB_:
```{r}
as.character(HBB_TF_names)
```

Search for these TFs in the GTEx gene expression list:
```{r}
for (i in HBB_TF_names){
    print(blood_gene_exp$Description[startsWith(blood_gene_exp$Description, as.character(i))] )
}
```

3 TFs _CACD_, _RAR_, and _HF1H3B_ can't be found in GTEx.

Extract expression data of the rest 13 TFs that are available in GTEx:
```{r}
HBB_TF_names <- HBB_TF_names[c(1,3,4,5,8:16)]
n = 13
HBB_TF_df <- data.frame(matrix(nrow = n, ncol = 407))
row.names(HBB_TF_df) <- HBB_TF_names
for (i in c(1:n)){
  HBB_TF_df[i,] <- blood_gene_exp[which(blood_gene_exp$Description==as.character(HBB_TF_names[i])),-1]
}

HBB_TF_df <- as.data.frame(t(HBB_TF_df))
HBB_exp <- blood_gene_exp[which(blood_gene_exp$Description=='HBB'),-1]
HBB_TF_df <- cbind(HBB_TF_df,t(HBB_exp))
colnames(HBB_TF_df) <- c(colnames(HBB_TF_df)[-14],'HBB')
```

Multiple linear regression on all 13 TFs:
```{r}
fit_all13 <- lm(HBB ~ ., data = HBB_TF_df)
summary(fit_all13)
```

Forward selection:
```{r echo=FALSE}
df <- HBB_TF_df
response_str <- "HBB ~ "
end <- length(colnames(df))
R <- colnames(df)[-end] # Initial set of predictors
pval <- rep(NA,length(R)); names(pval) <- R
R_sq <- rep(NA,length(R)); names(R_sq) <- R
for(j in R){
  fit <- lm(as.formula(paste0(response_str,j)), data=df)
  fitsum <- summary(fit)
  pval[j] <- fitsum$coefficients[j,4]
  R_sq[j] <- fitsum$r.squared
}
min_pval <- pval[which(pval == min(pval))]
r_sq <- R_sq[which(pval == min(pval))]
R[which(pval == min(pval))]
```

After the first predictor _GMEB2_ is selected,
add rest of the predictors one at a time until none of their p-values are significant (p value > 0.05).
```{r echo=TRUE}
stats <- data.frame(GMEB2 = c(min_pval,r_sq),row.names = c("p-val","R-squared"))
fit <- lm(as.formula(paste0(response_str,names(min_pval))), data=df)
R <- R[-which(R == names(min_pval))] # The rest of the predictors
nonstop = TRUE
while(nonstop){
  pval <- rep(NA,length(R)); names(pval) <- R
  R_sq <- rep(NA,length(R)); names(R_sq) <- R
  for(j in R){
    fit_loop <- update(fit, as.formula(paste0(".~. +",j)))
    fitsum <- summary(fit_loop)
    pval[j] <- fitsum$coefficients[j,4]
    R_sq[j] <- fitsum$r.squared
  }
  min_pval <- pval[which(pval == min(pval))]
  r_sq <- R_sq[which(pval == min(pval))]
  if(min_pval < 0.05){
    stats[,names(min_pval)] <- c(min_pval,r_sq)
    fit <- update(fit, as.formula(paste0(".~. +",names(min_pval))))
    R <- R[-which(R == names(min_pval))]
  }
  else{
    nonstop = FALSE
    forward_model <- fit
  }
}
```

The p-value and $R^2$ statistic for each of the significant predictors added at each round of selection:
```{r}
stats
```

Summary of the selected model:
```{r}
summary(forward_model)
```

```{r}
plot(as.matrix(stats)[2,], pch=16, xlab="Number of Predictors",ylab = "R_2")
```

