---
title: "TFBS ChIP-seq Data Analysis"
author: "Yifan Zhou"
date: "2/20/2018"
output:
  html_document:
    toc: true
---

```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.width=8, fig.height=6, fig.path='Figs/',
                      echo=TRUE, warning=FALSE, message=FALSE, comment = NA)
```

## ChIP-seq Data

TF binding site ChIP-seq data in LCL are downloaded from the [ENCODE Uniform TFBS track][ENCODE Uniform TFBS]:

* cell line: GM12878 (Tier 1)
* file format: [ENCODE narrowPeak][narrowPeak]

[ENCODE Uniform TFBS]: http://genome.ucsc.edu/cgi-bin/hgTrackUi?hgsid=657090723_G179QYBBXSSU1XvyOs35l9cq4BaC&c=chrY&g=wgEncodeAwgTfbsUniform
[narrowPeak]: https://genome.ucsc.edu/FAQ/FAQformat.html#format12

Open chromatin regions in LCL:

DHS peaks are available from [Grubert et al., 2015][Grubert] (Table S2), which were summarized over the original DHS profiles of 70 different LCL cell lines in [Degner et al., 2012][Degner].

* ~ 250,000 DHS peaks in total

[Grubert]: http://www.sciencedirect.com/science/article/pii/S0092867415009642?via%3Dihub#app1
[Degner]: https://www.nature.com/articles/nature10808#accessions

### TFs of interest: 

**YY1**

Encode ChIP-seq file name: `wgEncodeAwgTfbsSydhGm12878Yy1UniPk.narrowPeak`, integrated quality flag:	Caution.
  
**STAT1**

Encode ChIP-seq file name: `wgEncodeAwgTfbsSydhGm12878Stat1UniPk.narrowPeak`, integrated quality flag:	Caution.

**CEBPB**

Encode ChIP-seq file name: `wgEncodeAwgTfbsHaibGm12878Cebpbsc150V0422111UniPk.narrowPeak`, integrated quality flag:	Caution.

**SP1**

Encode ChIP-seq file name: `wgEncodeAwgTfbsHaibGm12878Sp1Pcr1xUniPk.narrowPeak`, integrated quality flag:	Good.

**NF-YA**

File name: `wgEncodeAwgTfbsSydhGm12878NfyaIggmusUniPk.narrowPeak`, integrated quality flag:	Good.


### Data Summary:

```{r echo=FALSE, results='asis'}
library(knitr)
tmp = data.frame(YY1 = c(2077, (1404*100)%/%2077, 1702, 1184, (554*100)%/%2077, 617), 
                 STAT1 = c(1769, (1483*100)%/%1769, 1970, 1603, (654*100)%/%1769, 727), 
                 CEBPB = c(5786, (5335*100)%/%5786, 6685, 2577, (899*100)%/%5786, 986), 
                 SP1 = c(18248, (17410*100)%/%18248, 18543, 9282, (6622*100)%/%18248, 6722), 
                 NFYA = c(1841, (1770*100)%/%1841, 1897, 2273, (1185*100)%/%1841, 1397),
                 row.names = c("# of ChIP-seq peaks","% of peaks in DHS","# of DHS overlapped","# of genes in contact w/ those DHS",
                               "% of peaks in 5kb gene promoters","# of 5kb gene promoters overlapped"))
kable(tmp) #, caption = "Data Summary")
```


## Analysis of Expression Correlation 

Analyze the correlation of mRNA expression levels btw a certain TF and all the other genes (one at a time), compare btw target genes & non-target genes:

* RNA-seq data: 16614 genes, 75 LCL cell lines ([Battle et al., 2015][Battle], Data Table S4).

[Battle]: http://science.sciencemag.org/content/suppl/2014/12/17/science.1260793.DC1?_ga=2.48064679.1826706411.1515443144-1380778862.1504648991

* Target genes are defined as those of which the +5kb promoter regions overlap with the TF ChIP-seq peaks.

### TF of interest: STAT1

[_STAT1_] ~ [_gene_] mRNA expression level correlation ($R^2$) distributions:

```{r include=FALSE}
hist_gen = function(R_sq,filename){
  target = read.table(filename, header = FALSE, na.strings = 'NA', stringsAsFactors = FALSE)
  
  seq = 1:length(R_sq$rsquare)
  target_order = match(target$V1, row.names(R_sq))
  nontarget_order = seq[-target_order]
  r_sq = R_sq$rsquare[target_order]
  if (length(r_sq[-which(r_sq==1)])>0){
    r_sq = r_sq[-which(r_sq==1)]
  }
  term = paste("Number of target genes:",length(r_sq))
  print(term)
  term = paste("Proportion among all genes:",round(length(r_sq)/length(R_sq$rsquare),digits = 4))
  print(term)
  hist_0 = hist(r_sq, plot = FALSE)
  hist_0$density = hist_0$counts/sum(hist_0$counts)*100
  
  non_r_sq = R_sq$rsquare[nontarget_order]
  if (length(non_r_sq[-which(non_r_sq==1)])>0){
    non_r_sq = non_r_sq[-which(non_r_sq==1)]
  }
  hist_1 = hist(non_r_sq, plot = FALSE)
  hist_1$density = hist_1$counts/sum(hist_1$counts)*100
  return(list(hist_0,hist_1))
}
```

* Target genes linked through ChIP-seq peaks with H3K27ac & in 3kb upstream gene promoter regions:

```{r echo=FALSE}
TF = "STAT1"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_0PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
```

```{r echo=FALSE}
filename = paste0('~/Downloads/Research/LCL/3kb_promoter_chipseq/',TF,'_UniPk_in_prom_w_H3K27ac_targets.txt')
hist = hist_gen(R_sq,filename)

title0 = paste0(TF," ~ Target-promoter genes")
title1 = paste0(TF," ~ Non-target-promoter genes")

par(mfrow=c(1,2))
plot(hist[[1]], freq = FALSE, main = title0, xlab = expression("R"^2))
plot(hist[[2]], freq = FALSE, main = title1, xlab = expression("R"^2))
```

```{r echo=FALSE}
title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.7), ylim = c(0,55), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
# mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
mtext("target genes: peaks overlap w/ 3kb promoter & H3K27ac", col = "gray30")
```


* After correcting for the top 2 PCs:

```{r echo=FALSE}
TF = "STAT1"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
```

```{r echo=FALSE}
filename = paste0('~/Downloads/Research/LCL/3kb_promoter_chipseq/',TF,'_UniPk_in_prom_w_H3K27ac_targets.txt')
hist = hist_gen(R_sq,filename)
title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 2PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.7), ylim = c(0,55), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
# mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
mtext("target genes: peaks overlap w/ 3kb promoter & H3K27ac", col = "gray30")
```

 *  Further pick the top 10% peaks based on their ChIP-se enrichment signal values, and link the target genes similarly through 3kb promoters (also corrected for the top 2 PCs):
 
```{r echo=FALSE}
TF = "STAT1"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
```

```{r echo=FALSE}
filename = paste0('~/Downloads/Research/LCL/3kb_promoter_chipseq/',TF,'_UniPk_top10_prom_targets.txt')
hist = hist_gen(R_sq,filename)
title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 2PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.7), ylim = c(0,55), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
# mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
mtext("target genes: top 10% peaks overlap w/ 3kb promoter & H3K27ac", col = "gray30")
```

 
 *  Target genes linked through ChIP-seq peaks in DHS+H3K27ac regions that are in Hi-C contact with gene promoters (also corrected for the top 2 PCs):

```{r echo=FALSE}
TF = "STAT1"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
```

```{r echo=FALSE}
filename = paste0('~/Downloads/Research/LCL/3kb_promoter_chipseq/',TF,'_UniPk_DHS_H3K27ac_genes.txt')
hist = hist_gen(R_sq,filename)
title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 2PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.7), ylim = c(0,55), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
mtext("target genes: linked by HiC through peaks in DHS & H3K27ac", col = "gray30")
# mtext("target genes: peaks overlap w/ 3kb promoter & H3K27ac", col = "gray30")
```

 *  Further pick the top 10% peaks based on their ChIP-se enrichment signal values, and link the target genes similarly through Hi-C (also corrected for the top 2 PCs):

```{r echo=FALSE}
TF = "STAT1"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
```

```{r echo=FALSE}
filename = paste0('~/Downloads/Research/LCL/3kb_promoter_chipseq/',TF,'_UniPk_top10_DHS_genes.txt')
hist = hist_gen(R_sq,filename)
title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 2PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.7), ylim = c(0,55), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
# mtext("target genes: peaks overlap w/ 3kb promoter & H3K27ac", col = "gray30")
```

* Peaks that contain STAT1 motif seqeunces from JASPAR:

```{r echo=FALSE}
TF = "STAT1"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))

filename = paste0('~/Downloads/Research/LCL/chipseq_w_motifs/',TF,'_peaks_w_motifs_3kb_prom.target')
hist = hist_gen(R_sq,filename)
title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 2PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.7), ylim = c(0,55), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
mtext("target genes: peaks w/ motifs & overlap w/ 3kb promoters", col = "gray30")
```

* Peaks that overlap with strong enhancers from ChromHmm, target genes linked through HiC:
```{r echo=FALSE}
TF = "STAT1"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
```

```{r echo=FALSE}
filename = paste0('~/Downloads/Research/LCL/DHS_chipseq/',TF,'_UniPk_StrEnh.target')
hist = hist_gen(R_sq,filename)
title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 2PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.7), ylim = c(0,60), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
# mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
mtext("target genes: linked by HiC through peaks that overlap w/ strong enhancers", col = "gray30")
```

```{r echo=FALSE}
TF = "STAT1"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_GD462.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
filename = paste0('~/Downloads/Research/LCL/DHS_chipseq/',TF,'_UniPk_StrEnh.GDtarget')
hist = hist_gen(R_sq,filename)
title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] GD462 R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.4), ylim = c(0,80), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
# mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
mtext("target genes: peaks overlap w/ 3kb promoters", col = "gray30")
```

```{r echo=FALSE}
TF = "STAT1"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_GD462.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
filename = paste0('~/Downloads/Research/LCL/3kb_promoter_chipseq/',TF,'_UniPk_GD_3kb_prom.target')
hist = hist_gen(R_sq,filename)
title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] GD462 R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.4), ylim = c(0,80), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
# mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
mtext("target genes: linked by HiC through peaks that overlap w/ strong enhancers", col = "gray30")
```

### TF of interest: YY1

[_YY1_] ~ [_gene_] mRNA expression level correlation ($R^2$) distributions:

```{r echo=FALSE}
TF = "YY1"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_0PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
filename = paste0('~/Downloads/Research/LCL/3kb_promoter_chipseq/',TF,'_UniPk_in_prom_w_H3K27ac_targets.txt')
hist = hist_gen(R_sq,filename)

title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 0PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.7), ylim = c(0,65), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
# mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
mtext("target genes: peaks overlap w/ 3kb promoter & H3K27ac", col = "gray30")
```

```{r echo=FALSE}
TF = "YY1"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
filename = paste0('~/Downloads/Research/LCL/3kb_promoter_chipseq/',TF,'_UniPk_in_prom_w_H3K27ac_targets.txt')
hist = hist_gen(R_sq,filename)

title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 2PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.7), ylim = c(0,65), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
# mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
mtext("target genes: peaks overlap w/ 3kb promoter & H3K27ac", col = "gray30")
```

```{r echo=FALSE}
TF = "YY1"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))

filename = paste0('~/Downloads/Research/LCL/DHS_chipseq/',TF,'_UniPk_StrEnh.target')
hist = hist_gen(R_sq,filename)
title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 2PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.7), ylim = c(0,65), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
# mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
mtext("target genes: linked by HiC through peaks that overlap w/ strong enhancers", col = "gray30")
```

```{r echo=FALSE}
TF = "YY1"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_GD462.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
filename = paste0('~/Downloads/Research/LCL/DHS_chipseq/',TF,'_UniPk_StrEnh.GDtarget')
hist = hist_gen(R_sq,filename)
title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] GD462 R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.2), ylim = c(0,80), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
# mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
mtext("target genes: linked by HiC through peaks that overlap w/ strong enhancers", col = "gray30")
```

### TF of interest: CEBPB

[_CEBPB_] ~ [_gene_] mRNA expression level correlation ($R^2$) distributions:

```{r echo=FALSE}
TF = "CEBPB"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_0PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
filename = paste0('~/Downloads/Research/LCL/3kb_promoter_chipseq/',TF,'_UniPk_in_prom_w_H3K27ac_targets.txt')
hist = hist_gen(R_sq,filename)

title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 0PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.6), ylim = c(0,75), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
# mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
mtext("target genes: peaks overlap w/ 3kb promoter & H3K27ac", col = "gray30")
```

```{r echo=FALSE}
TF = "CEBPB"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
filename = paste0('~/Downloads/Research/LCL/3kb_promoter_chipseq/',TF,'_UniPk_in_prom_w_H3K27ac_targets.txt')
hist = hist_gen(R_sq,filename)

title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 2PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.6), ylim = c(0,75), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
# mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
mtext("target genes: peaks overlap w/ 3kb promoter & H3K27ac", col = "gray30")
```

```{r echo=FALSE}
TF = "CEBPB"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_GD462.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
filename = paste0('~/Downloads/Research/LCL/DHS_chipseq/',TF,'_UniPk_StrEnh.GDtarget')
hist = hist_gen(R_sq,filename)
title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] GD462 R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.28), ylim = c(0,85), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
# mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
mtext("target genes: linked by HiC through peaks that overlap w/ strong enhancers", col = "gray30")
```

### TF of interest: SP1

[_Sp1_] ~ [_gene_] mRNA expression level correlation ($R^2$) distributions:

```{r echo=FALSE}
TF = "SP1"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_0PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
filename = paste0('~/Downloads/Research/LCL/3kb_promoter_chipseq/',TF,'_UniPk_3kb_prom.target')
hist = hist_gen(R_sq,filename)

title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 0PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.75), ylim = c(0,65), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
# mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
mtext("target genes: peaks overlap w/ 3kb promoter", col = "gray30")
```

```{r echo=FALSE}
TF = "SP1"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
filename = paste0('~/Downloads/Research/LCL/3kb_promoter_chipseq/',TF,'_UniPk_3kb_prom.target')
hist = hist_gen(R_sq,filename)

title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 2PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.75), ylim = c(0,65), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
# mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
mtext("target genes: peaks overlap w/ 3kb promoter", col = "gray30")
```

```{r echo=FALSE}
TF = "SP1"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
filename = paste0('~/Downloads/Research/LCL/chipseq_w_motifs/',TF,'_peaks_w_motifs_3kb_prom.target')
hist = hist_gen(R_sq,filename)

title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 2PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.75), ylim = c(0,65), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
mtext("target genes: peaks w/ motifs & overlap w/ 3kb promoter", col = "gray30")
```

```{r echo=FALSE}
TF = "SP1"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
filename = paste0('~/Downloads/Research/LCL/chipseq_w_motifs/',TF,'_peaks_w_motifs_HiC.target')
hist = hist_gen(R_sq,filename)

title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 2PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.75), ylim = c(0,65), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
mtext("target genes: linked by HiC through peaks w/ motifs and in DHS & H3K27ac", col = "gray30")
```

```{r echo=FALSE}
TF = "SP1"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))

filename = paste0('~/Downloads/Research/LCL/DHS_chipseq/',TF,'_UniPk_StrEnh.target')
hist = hist_gen(R_sq,filename)
title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 2PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.75), ylim = c(0,65), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
# mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
mtext("target genes: linked by HiC through peaks that overlap w/ strong enhancers", col = "gray30")
```

```{r echo=FALSE}
TF = "SP1"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_GD462.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
filename = paste0('~/Downloads/Research/LCL/DHS_chipseq/',TF,'_UniPk_StrEnh.GDtarget')
hist = hist_gen(R_sq,filename)
title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] GD462 R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.3), ylim = c(0,90), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
# mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
mtext("target genes: linked by HiC through peaks that overlap w/ strong enhancers", col = "gray30")
```

### TF of interest: NF-YA

[_NF-YA_] ~ [_gene_] mRNA expression level correlation ($R^2$) distributions:

```{r echo=FALSE}
TF = "NFYA"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_0PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
filename = paste0('~/Downloads/Research/LCL/3kb_promoter_chipseq/',TF,'_UniPk_3kb_prom.target')
hist = hist_gen(R_sq,filename)

title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 0PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.65), ylim = c(0,65), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
# mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
mtext("target genes: peaks overlap w/ 3kb promoter", col = "gray30")
```

```{r echo=FALSE}
TF = "NFYA"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
filename = paste0('~/Downloads/Research/LCL/3kb_promoter_chipseq/',TF,'_UniPk_3kb_prom.target')
hist = hist_gen(R_sq,filename)

title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 2PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.65), ylim = c(0,65), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
# mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
mtext("target genes: peaks overlap w/ 3kb promoter", col = "gray30")
```

```{r echo=FALSE}
TF = "NFYA"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))

filename = paste0('~/Downloads/Research/LCL/DHS_chipseq/',TF,'_UniPk_StrEnh.target')
hist = hist_gen(R_sq,filename)
title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 2PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.65), ylim = c(0,65), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
# mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
mtext("target genes: linked by HiC through peaks that overlap w/ strong enhancers", col = "gray30")
```

```{r echo=FALSE}
TF = "NFYA"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_GD462.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
filename = paste0('~/Downloads/Research/LCL/DHS_chipseq/',TF,'_UniPk_StrEnh.GDtarget')
hist = hist_gen(R_sq,filename)
title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] GD462 R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.2), ylim = c(0,70), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
# mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
mtext("target genes: linked by HiC through peaks that overlap w/ strong enhancers", col = "gray30")
```

### TF of interest: IRF4

[_IRF4_] ~ [_gene_] mRNA expression level correlation ($R^2$) distributions:

```{r echo=FALSE}
TF = "IRF4"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_0PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
filename = paste0('~/Downloads/Research/LCL/3kb_promoter_chipseq/',TF,'_UniPk_3kb_prom.target')
hist = hist_gen(R_sq,filename)

title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 0PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.6), ylim = c(0,70), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
mtext("target genes: peaks overlap w/ 3kb promoter", col = "gray30")
```

```{r echo=FALSE}
TF = "IRF4"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
filename = paste0('~/Downloads/Research/LCL/3kb_promoter_chipseq/',TF,'_UniPk_3kb_prom.target')
hist = hist_gen(R_sq,filename)

title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 2PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.6), ylim = c(0,70), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
mtext("target genes: peaks overlap w/ 3kb promoter", col = "gray30")
```

```{r echo=FALSE}
TF = "IRF4"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
filename = paste0('~/Downloads/Research/LCL/chipseq_w_motifs/',TF,'_peaks_w_motifs_3kb_prom.target')
hist = hist_gen(R_sq,filename)

title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 2PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.6), ylim = c(0,70), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
mtext("target genes: peaks w/ motifs & overlap w/ 3kb promoter", col = "gray30")
```

```{r echo=FALSE}
TF = "IRF4"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
filename = paste0('~/Downloads/Research/LCL/chipseq_w_motifs/',TF,'_peaks_w_motifs_HiC.target')
hist = hist_gen(R_sq,filename)

title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 2PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.6), ylim = c(0,70), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
mtext("target genes: linked by HiC through peaks w/ motifs and in DHS & H3K27ac", col = "gray30")
```

```{r echo=FALSE}
TF = "IRF4"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))

filename = paste0('~/Downloads/Research/LCL/DHS_chipseq/',TF,'_UniPk_StrEnh.target')
hist = hist_gen(R_sq,filename)
title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] (- 2PCs) R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.65), ylim = c(0,65), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
# mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
mtext("target genes: linked by HiC through peaks that overlap w/ strong enhancers", col = "gray30")
```

```{r echo=FALSE}
TF = "IRF4"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_GD462.txt')
R_sq = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
filename = paste0('~/Downloads/Research/LCL/DHS_chipseq/',TF,'_UniPk_StrEnh.GDtarget')
hist = hist_gen(R_sq,filename)
title = paste0("[",TF," ~ Target] vs [",TF,"~ Non-target] GD462 R2")
plot(hist[[1]]$mids, hist[[1]]$density, xlim = c(0,0.2), ylim = c(0,80), type = "o", col = "red",
     main = title, xlab = expression("R"^2), ylab = "Density (%)")
lines(hist[[2]]$mids, hist[[2]]$density,  type = "o", col = "blue")
# mtext("target genes: linked by HiC through top 10% peaks in DHS & H3K27ac", col = "gray30")
mtext("target genes: linked by HiC through peaks that overlap w/ strong enhancers", col = "gray30")
```
## New RNA expression level data: Geuvadis Database
* RNA-seq gene expression data: ([GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.gz][Geuvadis]).
* 23722 genes, 462 LCL cell lines from the 1000 Genomes
* Peer-factor normalized RPKM: library depth & transcript lengths & expressed & PEER
* 14102 genes overlap with Battle dataset

[Geuvadis]: https://www.ebi.ac.uk/arrayexpress/files/E-GEUV-1/analysis_results/

### $R^2$ difference between 2 datasets:
```{r echo=FALSE}
TF = "STAT1"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_GD462.txt')
R_sq_462 = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
tmp=R_sq_462[order(R_sq_462$rsquare,decreasing = T),]
names(tmp) = row.names(R_sq_462)[order(R_sq_462$rsquare,decreasing = T)]
tmp[1:5]

filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq_70 = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
tmp=R_sq_70[order(R_sq_70$rsquare,decreasing = T),]
names(tmp) = row.names(R_sq_70)[order(R_sq_70$rsquare,decreasing = T)]
tmp[1:5]

R_sq_70$rsquare_462 = R_sq_462$rsquare[match(row.names(R_sq_70),row.names(R_sq_462))]
tmp = na.omit(R_sq_70)
tmp = tmp[-which(row.names(tmp)=="ENSG00000115415"),]
title = paste(TF,"~ all genes R2 generated from 70 & 462 cell lines")
plot(1:length(tmp$rsquare),tmp$rsquare[order(tmp$rsquare)],type = "l", col="blue",main = title,
     xlab="genes in order",ylab=expression('R'^2*' values'))
lines(1:length(tmp$rsquare),tmp$rsquare_462[order(tmp$rsquare)], col="red")
legend(x=10000,y=0.6,lty=1,col=c("blue","red"),legend = c("Battle70,-2PCs","GD462"),cex=0.8)

sum_lm = summary(lm(tmp$rsquare~tmp$rsquare_462))
print(paste("Correlation:",sum_lm$r.squared))
```

```{r echo=FALSE}
TF = "YY1"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_GD462.txt')
R_sq_462 = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
tmp=R_sq_462[order(R_sq_462$rsquare,decreasing = T),]
names(tmp) = row.names(R_sq_462)[order(R_sq_462$rsquare,decreasing = T)]
tmp[1:5]

filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq_70 = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
tmp=R_sq_70[order(R_sq_70$rsquare,decreasing = T),]
names(tmp) = row.names(R_sq_70)[order(R_sq_70$rsquare,decreasing = T)]
tmp[1:5]

R_sq_70$rsquare_462 = R_sq_462$rsquare[match(row.names(R_sq_70),row.names(R_sq_462))]
tmp = na.omit(R_sq_70)
tmp = tmp[-which(row.names(tmp)=="ENSG00000100811"),]
title = paste(TF,"~ all genes R2 generated from 70 & 462 cell lines")
plot(1:length(tmp$rsquare),tmp$rsquare[order(tmp$rsquare)],type = "l", col="blue",main = title,
     xlab="genes in order",ylab=expression('R'^2*' values'))
lines(1:length(tmp$rsquare),tmp$rsquare_462[order(tmp$rsquare)], col="red")
legend(x=10000,y=0.6,lty=1,col=c("blue","red"),legend = c("Battle70,-2PCs","GD462"),cex=0.8)

sum_lm = summary(lm(tmp$rsquare~tmp$rsquare_462))
print(paste("Correlation:",sum_lm$r.squared))
```

```{r echo=FALSE}
TF = "CEBPB"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_GD462.txt')
R_sq_462 = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
tmp=R_sq_462[order(R_sq_462$rsquare,decreasing = T),]
names(tmp) = row.names(R_sq_462)[order(R_sq_462$rsquare,decreasing = T)]
tmp[1:5]

filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq_70 = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
tmp=R_sq_70[order(R_sq_70$rsquare,decreasing = T),]
names(tmp) = row.names(R_sq_70)[order(R_sq_70$rsquare,decreasing = T)]
tmp[1:5]

R_sq_70$rsquare_462 = R_sq_462$rsquare[match(row.names(R_sq_70),row.names(R_sq_462))]
tmp = na.omit(R_sq_70)
tmp = tmp[-which(row.names(tmp)=="ENSG00000172216"),]
title = paste(TF,"~ all genes R2 generated from 70 & 462 cell lines")
plot(1:length(tmp$rsquare),tmp$rsquare[order(tmp$rsquare)],type = "l", col="blue",main = title,
     xlab="genes in order",ylab=expression('R'^2*' values'))
lines(1:length(tmp$rsquare),tmp$rsquare_462[order(tmp$rsquare)], col="red")
legend(x=10000,y=0.6,lty=1,col=c("blue","red"),legend = c("Battle70,-2PCs","GD462"),cex=0.8)

sum_lm = summary(lm(tmp$rsquare~tmp$rsquare_462))
print(paste("Correlation:",sum_lm$r.squared))
```

```{r echo=FALSE}
TF = "SP1"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_GD462.txt')
R_sq_462 = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
tmp=R_sq_462[order(R_sq_462$rsquare,decreasing = T),]
names(tmp) = row.names(R_sq_462)[order(R_sq_462$rsquare,decreasing = T)]
tmp[1:5]

filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq_70 = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
tmp=R_sq_70[order(R_sq_70$rsquare,decreasing = T),]
names(tmp) = row.names(R_sq_70)[order(R_sq_70$rsquare,decreasing = T)]
tmp[1:5]

R_sq_70$rsquare_462 = R_sq_462$rsquare[match(row.names(R_sq_70),row.names(R_sq_462))]
tmp = na.omit(R_sq_70)
tmp = tmp[-which(row.names(tmp)=="ENSG00000185591"),]
title = paste(TF,"~ all genes R2 generated from 70 & 462 cell lines")
plot(1:length(tmp$rsquare),tmp$rsquare[order(tmp$rsquare)],type = "l", col="blue",main = title,
     xlab="genes in order",ylab=expression('R'^2*' values'))
lines(1:length(tmp$rsquare),tmp$rsquare_462[order(tmp$rsquare)], col="red")
legend(x=10000,y=0.7,lty=1,col=c("blue","red"),legend = c("Battle70,-2PCs","GD462"),cex=0.8)

sum_lm = summary(lm(tmp$rsquare~tmp$rsquare_462))
print(paste("Correlation:",sum_lm$r.squared))
```

```{r echo=FALSE}
TF = "NFYA"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_GD462.txt')
R_sq_462 = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
tmp=R_sq_462[order(R_sq_462$rsquare,decreasing = T),]
names(tmp) = row.names(R_sq_462)[order(R_sq_462$rsquare,decreasing = T)]
tmp[1:5]

filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq_70 = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
tmp=R_sq_70[order(R_sq_70$rsquare,decreasing = T),]
names(tmp) = row.names(R_sq_70)[order(R_sq_70$rsquare,decreasing = T)]
tmp[1:5]

R_sq_70$rsquare_462 = R_sq_462$rsquare[match(row.names(R_sq_70),row.names(R_sq_462))]
tmp = na.omit(R_sq_70)
tmp = tmp[-which(row.names(tmp)=="ENSG00000001167"),]
title = paste(TF,"~ all genes R2 generated from 70 & 462 cell lines")
plot(1:length(tmp$rsquare),tmp$rsquare[order(tmp$rsquare)],type = "l", col="blue",main = title,
     xlab="genes in order",ylab=expression('R'^2*' values'))
lines(1:length(tmp$rsquare),tmp$rsquare_462[order(tmp$rsquare)], col="red")
legend(x=10000,y=0.6,lty=1,col=c("blue","red"),legend = c("Battle70,-2PCs","GD462"),cex=0.8)

sum_lm = summary(lm(tmp$rsquare~tmp$rsquare_462))
print(paste("Correlation:",sum_lm$r.squared))
```

```{r echo=FALSE}
TF = "IRF4"
filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_GD462.txt')
R_sq_462 = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
tmp=R_sq_462[order(R_sq_462$rsquare,decreasing = T),]
names(tmp) = row.names(R_sq_462)[order(R_sq_462$rsquare,decreasing = T)]
tmp[1:5]

filename = paste0('~/Downloads/Research/LCL/',TF,'_all_genes_R2_2PCs.txt')
R_sq_70 = read.table(filename, sep = "\t", header = FALSE, row.names = 1, na.strings = 'NA', 
                  stringsAsFactors = FALSE, col.names = c("gene_id","rsquare"))
tmp=R_sq_70[order(R_sq_70$rsquare,decreasing = T),]
names(tmp) = row.names(R_sq_70)[order(R_sq_70$rsquare,decreasing = T)]
tmp[1:5]

R_sq_70$rsquare_462 = R_sq_462$rsquare[match(row.names(R_sq_70),row.names(R_sq_462))]
tmp = na.omit(R_sq_70)
tmp = tmp[-which(row.names(tmp)=="ENSG00000137265"),]
title = paste(TF,"~ all genes R2 generated from 70 & 462 cell lines")
plot(1:length(tmp$rsquare),tmp$rsquare[order(tmp$rsquare)],type = "l", col="blue",main = title,
     xlab="genes in order",ylab=expression('R'^2*' values'))
lines(1:length(tmp$rsquare),tmp$rsquare_462[order(tmp$rsquare)], col="red")
legend(x=10000,y=0.5,lty=1,col=c("blue","red"),legend = c("Battle70,-2PCs","GD462"),cex=0.8)

sum_lm = summary(lm(tmp$rsquare~tmp$rsquare_462))
print(paste("Correlation:",sum_lm$r.squared))
```

