<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Yifan Zhou" />


<title>HBB-TF Regression Analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}

.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Gene Regulation</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Description</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Progress
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="enhancer_tss.html">Enhancer-TSS</a>
    </li>
    <li>
      <a href="prelim_TF_gene_analysis.html">TF-Gene(LY75)</a>
    </li>
    <li>
      <a href="HBB-TF_Regression_Analysis.html">TF-Gene(HBB)</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">HBB-TF Regression Analysis</h1>
<h4 class="author"><em>Yifan Zhou</em></h4>

</div>


<div id="load-data" class="section level2">
<h2>Load Data</h2>
<p>We choose whole blood as the tissue of interest for preliminary analysis.</p>
<p>Load tissue-specific (whole blood) gene expression tpm data extracted from GTEx v7 <code>GTEx_Analysis_2016-01-15_v7_RNASeQCv1.1.8_gene_tpm.gct</code>, and do normalization:</p>
<pre class="r"><code>load(&#39;~/Downloads/Research/GTEx/blood_gene_exp_norm.Rdata&#39;)</code></pre>
<p>Load data from ‘tissue-specific circuits’ paper (<code>blood_adult.txt</code> from [394 individual networks] <a href="http://regulatorycircuits.org/download.html" class="uri">http://regulatorycircuits.org/download.html</a>):</p>
<pre class="r"><code>blood_TF_gene &lt;- read.delim(file = &#39;~/Downloads/Research/GTEx/blood_adult_TF_gene.txt&#39;,
                            sep = &#39;\t&#39;, header = FALSE, stringsAsFactors = FALSE) </code></pre>
</div>
<div id="hbb-and-its-associated-tfs" class="section level2">
<h2><em>HBB</em> and its Associated TFs</h2>
<p>We choose <em>HBB</em>, a gene that encodes haemoglobin subunit beta as a target gene of interest to analyze the association of <em>HBB</em> expression level with the expression levels of putative TFs linked to <em>HBB</em>:</p>
<pre class="r"><code>HBB &lt;- blood_TF_gene[which(blood_TF_gene[,2]==&#39;HBB&#39;),]
HBB &lt;- HBB[order(HBB[,3],decreasing = TRUE),]
HBB_TF_names&lt;- HBB[,1]</code></pre>
<p>Putative TF list for gene <em>HBB</em>:</p>
<pre class="r"><code>as.character(HBB_TF_names)</code></pre>
<pre><code> [1] &quot;KLF4&quot;   &quot;CACD&quot;   &quot;GATA3&quot;  &quot;GATA4&quot;  &quot;GATA5&quot;  &quot;RAR&quot;    &quot;HF1H3B&quot;
 [8] &quot;SP1&quot;    &quot;ZBTB33&quot; &quot;GATA6&quot;  &quot;ATF2&quot;   &quot;GTF2I&quot;  &quot;NFIC&quot;   &quot;NFIL3&quot; 
[15] &quot;HNF4A&quot;  &quot;GMEB2&quot; </code></pre>
<p>Search for these 16 TFs in the GTEx gene expression list:</p>
<pre class="r"><code>for (i in HBB_TF_names){
    print(blood_gene_exp_norm$Description[startsWith(blood_gene_exp_norm$Description, as.character(i))] )
}</code></pre>
<pre><code>[1] &quot;KLF4&quot;
character(0)
[1] &quot;GATA3-AS1&quot; &quot;GATA3&quot;    
[1] &quot;GATA4&quot;
[1] &quot;GATA5&quot;
 [1] &quot;RARB&quot;      &quot;RARRES2P1&quot; &quot;RARRES1&quot;   &quot;RARRES2P4&quot; &quot;RARS&quot;     
 [6] &quot;RARS2&quot;     &quot;RARRES2&quot;   &quot;RARRES2P2&quot; &quot;RARRES3&quot;   &quot;RARG&quot;     
[11] &quot;RARRES2P5&quot; &quot;RARRES2P6&quot; &quot;RARRES2P9&quot; &quot;RARRES2P7&quot; &quot;RARRES2P8&quot;
[16] &quot;RARA&quot;      &quot;RARRES2P3&quot;
character(0)
[1] &quot;SP110&quot;  &quot;SP140&quot;  &quot;SP140L&quot; &quot;SP100&quot;  &quot;SP1&quot;   
[1] &quot;ZBTB33&quot;
[1] &quot;GATA6-AS1&quot; &quot;GATA6&quot;    
[1] &quot;ATF2&quot;
[1] &quot;GTF2IRD1P1&quot; &quot;GTF2IRD2P1&quot; &quot;GTF2IRD1&quot;   &quot;GTF2I&quot;      &quot;GTF2IRD2&quot;  
[6] &quot;GTF2IRD2B&quot;  &quot;GTF2IP1&quot;    &quot;GTF2IP3&quot;    &quot;GTF2IP2&quot;   
[1] &quot;NFIC&quot;
[1] &quot;NFIL3&quot;
[1] &quot;HNF4A&quot;
[1] &quot;GMEB2&quot;</code></pre>
<p>3 TFs, <em>CACD</em>, <em>RAR</em>, and <em>HF1H3B</em>, cannot be found in GTEx.</p>
<p>Extract expression data of the rest 13 TFs that are available in GTEx:</p>
<pre class="r"><code>HBB_TF_names &lt;- HBB_TF_names[c(1,3,4,5,8:16)]
n = 13
HBB_TF_exp &lt;- data.frame(matrix(nrow = 407, ncol = n))
for (i in c(1:n)){
  HBB_TF_exp[,i] &lt;- t(blood_gene_exp_norm[which(blood_gene_exp_norm$Description
                                         ==as.character(HBB_TF_names[i])),-1])
}

HBB_exp &lt;- blood_gene_exp_norm[which(blood_gene_exp_norm$Description==&#39;HBB&#39;),-1]
HBB_TF_exp &lt;- cbind(HBB_TF_exp,t(HBB_exp))
colnames(HBB_TF_exp) &lt;- c(HBB_TF_names,&#39;HBB&#39;)
row.names(HBB_TF_exp) &lt;- colnames(blood_gene_exp_norm)[-1]</code></pre>
</div>
<div id="linear-regression" class="section level2">
<h2>Linear Regression</h2>
<p>Multiple linear regression on all 13 TFs:</p>
<pre class="r"><code>fit_all_HBB &lt;- lm(HBB ~ ., data = HBB_TF_exp)
summary(fit_all_HBB)</code></pre>
<pre><code>
Call:
lm(formula = HBB ~ ., data = HBB_TF_exp)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.222574 -0.055046  0.001484  0.051732  0.224473 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)  5.310e-01  9.783e-03  54.279  &lt; 2e-16 ***
KLF4         1.782e+03  3.680e+02   4.842 1.85e-06 ***
GATA3       -1.752e+03  1.093e+03  -1.602   0.1099    
GATA4       -4.962e+04  3.735e+04  -1.329   0.1848    
GATA5        5.630e+03  3.789e+04   0.149   0.8819    
SP1         -1.950e+03  2.925e+02  -6.667 8.88e-11 ***
ZBTB33       4.463e+03  3.401e+03   1.312   0.1902    
GATA6       -3.088e+04  1.330e+04  -2.322   0.0207 *  
ATF2        -2.579e+03  2.181e+03  -1.183   0.2377    
GTF2I       -1.818e+03  2.517e+03  -0.723   0.4704    
NFIC        -9.473e+03  1.883e+03  -5.031 7.42e-07 ***
NFIL3       -3.714e+01  3.313e+01  -1.121   0.2630    
HNF4A       -2.238e+04  1.179e+04  -1.898   0.0584 .  
GMEB2       -8.123e+03  1.119e+03  -7.262 2.06e-12 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.07884 on 393 degrees of freedom
Multiple R-squared:  0.7077,    Adjusted R-squared:  0.698 
F-statistic: 73.18 on 13 and 393 DF,  p-value: &lt; 2.2e-16</code></pre>
<p><strong>Forward Selection</strong></p>
<pre><code>[1] &quot;GMEB2&quot;</code></pre>
<p>After the first predictor <em>GMEB2</em> is selected, add rest of the predictors one at a time until none of their p-values are significant (p value &gt; 0.05).</p>
<pre class="r"><code>stats &lt;- data.frame(GMEB2 = c(min_pval,r_sq),row.names = c(&quot;p-val&quot;,&quot;R-squared&quot;))
fit &lt;- lm(as.formula(paste0(response_str,names(min_pval))), data=df)
R &lt;- R[-which(R == names(min_pval))] # The rest of the predictors
nonstop = TRUE
while(nonstop){
  pval &lt;- rep(NA,length(R)); names(pval) &lt;- R
  R_sq &lt;- rep(NA,length(R)); names(R_sq) &lt;- R
  for(j in R){
    fit_loop &lt;- update(fit, as.formula(paste0(&quot;.~. +&quot;,j)))
    fitsum &lt;- summary(fit_loop)
    pval[j] &lt;- fitsum$coefficients[j,4]
    R_sq[j] &lt;- fitsum$r.squared
  }
  min_pval &lt;- pval[which(pval == min(pval))]
  r_sq &lt;- R_sq[which(pval == min(pval))]
  if(min_pval &lt; 0.05){
    stats[,names(min_pval)] &lt;- c(min_pval,r_sq)
    fit &lt;- update(fit, as.formula(paste0(&quot;.~. +&quot;,names(min_pval))))
    R &lt;- R[-which(R == names(min_pval))]
  }
  else{
    nonstop = FALSE
    forward_model &lt;- fit
  }
}
rm(list = &#39;fit_loop&#39;)</code></pre>
<p>The p-value and <span class="math inline">\(R^2\)</span> statistic for each of the significant predictors added at each round of selection:</p>
<pre class="r"><code>stats</code></pre>
<pre><code>                 GMEB2          SP1        GATA6         NFIC         KLF4
p-val     6.496898e-66 2.329434e-29 1.261419e-05 0.0009530469 1.268444e-06
R-squared 5.165987e-01 6.467329e-01 6.630782e-01 0.6721155521 6.907782e-01
                HNF4A      NFIL3      GATA3
p-val     0.003112527 0.01712368 0.04037568
R-squared 0.697469941 0.70175418 0.70489036</code></pre>
<p>Summary of the selected model:</p>
<pre class="r"><code>summary(forward_model)</code></pre>
<pre><code>
Call:
lm(formula = HBB ~ GMEB2 + SP1 + GATA6 + NFIC + KLF4 + HNF4A + 
    NFIL3 + GATA3, data = df)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.223088 -0.054531  0.002897  0.052857  0.228688 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)  5.290e-01  9.682e-03  54.640  &lt; 2e-16 ***
GMEB2       -7.945e+03  1.101e+03  -7.217 2.71e-12 ***
SP1         -2.015e+03  2.323e+02  -8.675  &lt; 2e-16 ***
GATA6       -3.739e+04  9.398e+03  -3.978 8.24e-05 ***
NFIC        -9.457e+03  1.807e+03  -5.235 2.68e-07 ***
KLF4         1.747e+03  3.473e+02   5.031 7.41e-07 ***
HNF4A       -2.879e+04  1.063e+04  -2.707  0.00708 ** 
NFIL3       -6.397e+01  2.747e+01  -2.328  0.02040 *  
GATA3       -1.808e+03  8.792e+02  -2.057  0.04038 *  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.07871 on 398 degrees of freedom
Multiple R-squared:  0.7049,    Adjusted R-squared:  0.699 
F-statistic: 118.8 on 8 and 398 DF,  p-value: &lt; 2.2e-16</code></pre>
<pre class="r"><code>plot(as.matrix(stats)[2,], pch=16, xlab=&quot;Number of Predictors&quot;,ylab = &quot;R_2&quot;)</code></pre>
<p><img src="Figs/unnamed-chunk-12-1.png" width="576" /></p>
<p>Check the normality assumption:</p>
<pre class="r"><code>par(mfrow=c(1,2))
hist(forward_model$residuals)
qqnorm(forward_model$residuals,pch=16) 
qqline(forward_model$residuals,col=&#39;red&#39;)</code></pre>
<p><img src="Figs/unnamed-chunk-13-1.png" width="576" /></p>
<p><strong>Bayesian Information Criterion (BIC)</strong></p>
<pre class="r"><code>library(leaps)
RSS &lt;- summary(regsubsets(HBB ~., HBB_TF_exp)) 
n &lt;- dim(HBB_TF_exp)[1]
p &lt;- length(RSS$rss)+1
BIC &lt;- n*log(RSS$rss/n) + log(n)*c(2:p) 
plot(BIC,xlab=&#39;Number of Predictors&#39;, pch=16)</code></pre>
<p><img src="Figs/unnamed-chunk-14-1.png" width="576" /></p>
<p>BIC is maximized when the following 6 predictors are used:</p>
<pre class="r"><code>BIC_predictors &lt;- RSS$which[6,RSS$which[6,]]
BIC_predictors &lt;- names(BIC_predictors)[-1]
BIC_predictors</code></pre>
<pre><code>[1] &quot;KLF4&quot;  &quot;SP1&quot;   &quot;GATA6&quot; &quot;NFIC&quot;  &quot;HNF4A&quot; &quot;GMEB2&quot;</code></pre>
<p>BIC selected model:</p>
<pre class="r"><code>BIC_model &lt;- lm(HBB  ~ KLF4 + SP1 + GATA6 + NFIC + HNF4A + GMEB2, data = HBB_TF_exp)
summary(BIC_model)</code></pre>
<pre><code>
Call:
lm(formula = HBB ~ KLF4 + SP1 + GATA6 + NFIC + HNF4A + GMEB2, 
    data = HBB_TF_exp)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.224867 -0.056064  0.001213  0.051060  0.236599 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)  5.207e-01  9.405e-03  55.365  &lt; 2e-16 ***
KLF4         1.659e+03  3.381e+02   4.905 1.36e-06 ***
SP1         -1.851e+03  2.092e+02  -8.849  &lt; 2e-16 ***
GATA6       -4.451e+04  9.057e+03  -4.914 1.30e-06 ***
NFIC        -9.730e+03  1.822e+03  -5.339 1.57e-07 ***
HNF4A       -3.168e+04  1.065e+04  -2.974  0.00311 ** 
GMEB2       -9.351e+03  9.422e+02  -9.924  &lt; 2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.0795 on 400 degrees of freedom
Multiple R-squared:  0.6975,    Adjusted R-squared:  0.6929 
F-statistic: 153.7 on 6 and 400 DF,  p-value: &lt; 2.2e-16</code></pre>
<p><strong>Normality assumption</strong></p>
<pre class="r"><code>par(mfrow=c(1,2))
hist(BIC_model$residuals)
qqnorm(BIC_model$residuals,pch=16) 
qqline(BIC_model$residuals,col=&#39;red&#39;)</code></pre>
<p><img src="Figs/unnamed-chunk-17-1.png" width="576" /></p>
</div>
<div id="negative-controls" class="section level2">
<h2>Negative Controls</h2>
<p><span class="math inline">\(R^2\)</span>s from the control models:</p>
<pre class="r"><code>hist(R_sq_HBB, main = &#39;Distribution of R_sq of Control Models&#39;)</code></pre>
<p><img src="Figs/unnamed-chunk-19-1.png" width="576" /></p>
<pre class="r"><code>summary(R_sq_HBB)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
0.00000 0.00000 0.00000 0.01566 0.00000 0.81316 </code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
