<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Yifan Zhou" />


<title>Gene-TF Regression Analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}

.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Gene Regulation</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Progress
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="enhancer_tss.html">Enhancer-TSS</a>
    </li>
    <li>
      <a href="HBB-TF_Regression_Analysis.html">TF-Gene</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Gene-TF Regression Analysis</h1>
<h4 class="author"><em>Yifan Zhou</em></h4>

</div>


<p>We choose whole blood as the tissue of interest for preliminary analysis.</p>
<p>Load tissue-specific (whole blood) gene expression tpm data extraced from GTEx v7 <code>GTEx_Analysis_2016-01-15_v7_RNASeQCv1.1.8_gene_tpm.gct</code>:</p>
<pre class="r"><code>blood_gene_exp &lt;- read.delim(file = &#39;Data/blood_gene_expression.txt&#39;,
                             sep = &#39;\t&#39;,header = TRUE,stringsAsFactors = FALSE)</code></pre>
<p>Load data from ‘tissue-specific circuits’ paper (<code>blood_adult.txt</code> from [394 individual networks] <a href="http://regulatorycircuits.org/download.html" class="uri">http://regulatorycircuits.org/download.html</a>):</p>
<pre class="r"><code>blood_TF_gene &lt;- read.delim(file = &#39;Data/blood_adult_TF_gene.txt&#39;,sep = &#39;\t&#39;,
                            header = FALSE,stringsAsFactors = TRUE) 
gene_names &lt;-blood_TF_gene[,2]</code></pre>
<p>We choose <em>HBB</em>, a gene that encodes haemoglobin subunit beta as a target gene of interest to analyze the association of <em>HBB</em> expression level with the expression levels of putative TFs linked to <em>HBB</em>:</p>
<pre class="r"><code>HBB &lt;- blood_TF_gene[which(gene_names==&#39;HBB&#39;),]
temp &lt;- HBB[,3]
HBB &lt;- HBB[order(temp,decreasing = TRUE),]
HBB_TF_names&lt;- HBB[,1]</code></pre>
<p>Putative TF list for gene <em>HBB</em>:</p>
<pre class="r"><code>as.character(HBB_TF_names)</code></pre>
<pre><code> [1] &quot;KLF4&quot;   &quot;CACD&quot;   &quot;GATA3&quot;  &quot;GATA4&quot;  &quot;GATA5&quot;  &quot;RAR&quot;    &quot;HF1H3B&quot;
 [8] &quot;SP1&quot;    &quot;ZBTB33&quot; &quot;GATA6&quot;  &quot;ATF2&quot;   &quot;GTF2I&quot;  &quot;NFIC&quot;   &quot;NFIL3&quot; 
[15] &quot;HNF4A&quot;  &quot;GMEB2&quot; </code></pre>
<p>Search for these TFs in the GTEx gene expression list:</p>
<pre class="r"><code>for (i in HBB_TF_names){
    print(blood_gene_exp$Description[startsWith(blood_gene_exp$Description, as.character(i))] )
}</code></pre>
<pre><code>[1] &quot;KLF4&quot;
character(0)
[1] &quot;GATA3-AS1&quot; &quot;GATA3&quot;    
[1] &quot;GATA4&quot;
[1] &quot;GATA5&quot;
 [1] &quot;RARB&quot;      &quot;RARRES2P1&quot; &quot;RARRES1&quot;   &quot;RARRES2P4&quot; &quot;RARS&quot;     
 [6] &quot;RARS2&quot;     &quot;RARRES2&quot;   &quot;RARRES2P2&quot; &quot;RARRES3&quot;   &quot;RARG&quot;     
[11] &quot;RARRES2P5&quot; &quot;RARRES2P6&quot; &quot;RARRES2P9&quot; &quot;RARRES2P7&quot; &quot;RARRES2P8&quot;
[16] &quot;RARA&quot;      &quot;RARRES2P3&quot;
character(0)
[1] &quot;SP110&quot;  &quot;SP140&quot;  &quot;SP140L&quot; &quot;SP100&quot;  &quot;SP1&quot;   
[1] &quot;ZBTB33&quot;
[1] &quot;GATA6-AS1&quot; &quot;GATA6&quot;    
[1] &quot;ATF2&quot;
[1] &quot;GTF2IRD1P1&quot; &quot;GTF2IRD2P1&quot; &quot;GTF2IRD1&quot;   &quot;GTF2I&quot;      &quot;GTF2IRD2&quot;  
[6] &quot;GTF2IRD2B&quot;  &quot;GTF2IP1&quot;    &quot;GTF2IP3&quot;    &quot;GTF2IP2&quot;   
[1] &quot;NFIC&quot;
[1] &quot;NFIL3&quot;
[1] &quot;HNF4A&quot;
[1] &quot;GMEB2&quot;</code></pre>
<p>3 TFs <em>CACD</em>, <em>RAR</em>, and <em>HF1H3B</em> can’t be found in GTEx.</p>
<p>Extract expression data of the rest 13 TFs that are available in GTEx:</p>
<pre class="r"><code>HBB_TF_names &lt;- HBB_TF_names[c(1,3,4,5,8:16)]
n = 13
HBB_TF_df &lt;- data.frame(matrix(nrow = n, ncol = 407))
row.names(HBB_TF_df) &lt;- HBB_TF_names
for (i in c(1:n)){
  HBB_TF_df[i,] &lt;- blood_gene_exp[which(blood_gene_exp$Description==as.character(HBB_TF_names[i])),-1]
}

HBB_TF_df &lt;- as.data.frame(t(HBB_TF_df))
HBB_exp &lt;- blood_gene_exp[which(blood_gene_exp$Description==&#39;HBB&#39;),-1]
HBB_TF_df &lt;- cbind(HBB_TF_df,t(HBB_exp))
colnames(HBB_TF_df) &lt;- c(colnames(HBB_TF_df)[-14],&#39;HBB&#39;)</code></pre>
<p>Multiple linear regression on all 13 TFs:</p>
<pre class="r"><code>fit_all13 &lt;- lm(HBB ~ ., data = HBB_TF_df)
summary(fit_all13)</code></pre>
<pre><code>
Call:
lm(formula = HBB ~ ., data = HBB_TF_df)

Residuals:
    Min      1Q  Median      3Q     Max 
-222583  -55048    1499   51723  224489 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) 530994.00    9782.82  54.278  &lt; 2e-16 ***
KLF4          1782.06     368.04   4.842 1.85e-06 ***
GATA3        -1751.75    1093.45  -1.602   0.1100    
GATA4       -49612.95   37347.67  -1.328   0.1848    
GATA5         5630.12   37887.24   0.149   0.8819    
SP1          -1950.25     292.54  -6.667 8.88e-11 ***
ZBTB33        4462.75    3400.92   1.312   0.1902    
GATA6       -30880.20   13296.20  -2.322   0.0207 *  
ATF2         -2579.34    2181.26  -1.183   0.2377    
GTF2I        -1818.23    2516.68  -0.722   0.4704    
NFIC         -9472.83    1882.73  -5.031 7.42e-07 ***
NFIL3          -37.14      33.13  -1.121   0.2630    
HNF4A       -22384.84   11791.06  -1.898   0.0584 .  
GMEB2        -8122.76    1118.55  -7.262 2.06e-12 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 78840 on 393 degrees of freedom
Multiple R-squared:  0.7077,    Adjusted R-squared:  0.698 
F-statistic: 73.18 on 13 and 393 DF,  p-value: &lt; 2.2e-16</code></pre>
<p>Forward selection:</p>
<pre><code>[1] &quot;GMEB2&quot;</code></pre>
<p>After the first predictor <em>GMEB2</em> is selected, add rest of the predictors one at a time until none of their p-values are significant (p value &gt; 0.05).</p>
<pre class="r"><code>stats &lt;- data.frame(GMEB2 = c(min_pval,r_sq),row.names = c(&quot;p-val&quot;,&quot;R-squared&quot;))
fit &lt;- lm(as.formula(paste0(response_str,names(min_pval))), data=df)
R &lt;- R[-which(R == names(min_pval))] # The rest of the predictors
nonstop = TRUE
while(nonstop){
  pval &lt;- rep(NA,length(R)); names(pval) &lt;- R
  R_sq &lt;- rep(NA,length(R)); names(R_sq) &lt;- R
  for(j in R){
    fit_loop &lt;- update(fit, as.formula(paste0(&quot;.~. +&quot;,j)))
    fitsum &lt;- summary(fit_loop)
    pval[j] &lt;- fitsum$coefficients[j,4]
    R_sq[j] &lt;- fitsum$r.squared
  }
  min_pval &lt;- pval[which(pval == min(pval))]
  r_sq &lt;- R_sq[which(pval == min(pval))]
  if(min_pval &lt; 0.05){
    stats[,names(min_pval)] &lt;- c(min_pval,r_sq)
    fit &lt;- update(fit, as.formula(paste0(&quot;.~. +&quot;,names(min_pval))))
    R &lt;- R[-which(R == names(min_pval))]
  }
  else{
    nonstop = FALSE
    forward_model &lt;- fit
  }
}</code></pre>
<p>The p-value and <span class="math inline">\(R^2\)</span> statistic for each of the significant predictors added at each round of selection:</p>
<pre class="r"><code>stats</code></pre>
<pre><code>                 GMEB2          SP1        GATA6         NFIC         KLF4
p-val     6.503978e-66 2.334295e-29 1.261083e-05 0.0009524463 1.269002e-06
R-squared 5.165961e-01 6.467274e-01 6.630734e-01 0.6721117985 6.907740e-01
                HNF4A      NFIL3     GATA3
p-val     0.003113463 0.01712264 0.0403808
R-squared 0.697465423 0.70174980 0.7048859</code></pre>
<p>Summary of the selected model:</p>
<pre class="r"><code>summary(forward_model)</code></pre>
<pre><code>
Call:
lm(formula = HBB ~ GMEB2 + SP1 + GATA6 + NFIC + KLF4 + HNF4A + 
    NFIL3 + GATA3, data = df)

Residuals:
    Min      1Q  Median      3Q     Max 
-223086  -54536    2887   52848  228704 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) 529040.94    9682.31  54.640  &lt; 2e-16 ***
GMEB2        -7944.89    1100.82  -7.217 2.71e-12 ***
SP1          -2015.45     232.34  -8.675  &lt; 2e-16 ***
GATA6       -37386.56    9397.72  -3.978 8.25e-05 ***
NFIC         -9457.63    1806.61  -5.235 2.68e-07 ***
KLF4          1747.02     347.27   5.031 7.41e-07 ***
HNF4A       -28790.25   10635.29  -2.707  0.00708 ** 
NFIL3          -63.97      27.47  -2.328  0.02040 *  
GATA3        -1808.20     879.24  -2.057  0.04038 *  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 78710 on 398 degrees of freedom
Multiple R-squared:  0.7049,    Adjusted R-squared:  0.699 
F-statistic: 118.8 on 8 and 398 DF,  p-value: &lt; 2.2e-16</code></pre>
<pre class="r"><code>plot(as.matrix(stats)[2,], pch=16, xlab=&quot;Number of Predictors&quot;,ylab = &quot;R_2&quot;)</code></pre>
<p><img src="Figs/unnamed-chunk-12-1.png" width="576" /></p>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
