<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Yifan Zhou" />


<title>Preliminary Analysis of TF-Gene Association</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}

.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Gene Regulation</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Description</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Progress
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="enhancer_tss.html">Enhancer-TSS</a>
    </li>
    <li>
      <a href="prelim_TF_gene_analysis.html">TF-Gene(LY75)</a>
    </li>
    <li>
      <a href="HBB-TF_Regression_Analysis.html">TF-Gene(HBB)</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Preliminary Analysis of TF-Gene Association</h1>
<h4 class="author"><em>Yifan Zhou</em></h4>

</div>


<div id="load-data" class="section level2">
<h2>Load data</h2>
<p>We choose whole blood as the tissue of interest for preliminary analysis.</p>
<p>Load tissue-specific (whole blood) gene expression tpm data extracted from GTEx v7 <code>GTEx_Analysis_2016-01-15_v7_RNASeQCv1.1.8_gene_tpm.gct</code>, and do normalization:</p>
<pre class="r"><code>load(&#39;~/Downloads/Research/GTEx/blood_gene_exp_norm.Rdata&#39;)</code></pre>
<p>Load data from results of ‘tissue-specific circuits’ paper (<code>blood_adult.txt</code> from one of the 394 individual networks they generated, downloaded from <a href="http://regulatorycircuits.org/download.html" class="uri">http://regulatorycircuits.org/download.html</a>):</p>
<pre class="r"><code>blood_TF_gene &lt;- read.delim(file = &#39;~/Downloads/Research/GTEx/blood_adult_TF_gene.txt&#39;,
                            sep = &#39;\t&#39;, header = FALSE, stringsAsFactors = FALSE) </code></pre>
<p><strong>GTEx expression data overview</strong></p>
<pre class="r"><code>sum_gene &lt;- rowSums(blood_gene_exp_norm[,-1])
names(sum_gene) &lt;- blood_gene_exp_norm$Description
summary(sum_gene)</code></pre>
<pre><code>    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
0.00e+00 0.00e+00 1.00e-05 7.24e-03 3.20e-04 1.08e+02 </code></pre>
<pre class="r"><code>hist(log10(sum_gene))</code></pre>
<p><img src="Figs/unnamed-chunk-4-1.png" width="576" /></p>
<p>The distribution of gene expression is roughly log-normal. A few genes have extremely high expression rate.</p>
<p>Top 30 expressed genes in blood tissue type:</p>
<pre class="r"><code>top30 &lt;- sum_gene[which(log10(sum_gene)&gt;0)]
top30</code></pre>
<pre><code>     CSF3R    S100A11     S100A9    S100A12     S100A8      HLA-A 
  1.089221   1.779593  15.250304   1.291707   1.756228   1.347485 
     HLA-E      HLA-C      HLA-B       ACTB       SRGN     IFITM2 
  1.220447   2.301403   3.677534   2.282394   1.193987   3.272747 
       HBB        HBD       FTH1        B2M       HBA2       HBA1 
107.958238   6.987105   1.650927   1.205410  36.407871   7.295904 
       FTL     TMSB4X    MT-RNR2     MT-ND1     MT-ND2     MT-CO1 
  5.729424   1.153190   2.004031   1.292850   1.070329   1.945645 
    MT-CO2    MT-ATP8    MT-ATP6     MT-CO3     MT-ND4     MT-CYB 
  2.490420   1.184855   2.185321   2.039552   2.064072   1.184578 </code></pre>
</div>
<div id="ly75-and-its-associated-tfs" class="section level2">
<h2><em>LY75</em> and its associated TFs</h2>
<p><em>LY75</em> is a gene that encodes DEC-205 (also known as CD205), a cell surface receptor that is highly expressed on dendritic cells and is involved in the endocytosis of extracellular antigens and their presentation on MHC class I molecules. We choose <em>LY75</em> as a target gene of interest to analyze the association of <em>LY75</em> expression level with the expression levels of putative TFs linked to <em>LY75</em>:</p>
<p>We first identify a list of putative TFs that are linked to <em>LY75</em> according to results from the paper:</p>
<pre class="r"><code>LY75 &lt;- blood_TF_gene[which(blood_TF_gene[,2]==&#39;LY75&#39;),]
LY75 &lt;- LY75[order(LY75[,3],decreasing = TRUE),]
LY75_TF_names&lt;- LY75[,1]
t(LY75_TF_names)</code></pre>
<pre><code>     [,1]  [,2]     [,3]     [,4]    [,5]    [,6]  [,7]   [,8]    
[1,] &quot;SP1&quot; &quot;HF1H3B&quot; &quot;TFAP2A&quot; &quot;HIF1A&quot; &quot;CREB1&quot; &quot;AHR&quot; &quot;ARNT&quot; &quot;ZNF75A&quot;
     [,9]     [,10]  [,11]    [,12]    [,13]    [,14]   [,15]   [,16]   
[1,] &quot;TFAP2C&quot; &quot;ETV6&quot; &quot;HOXD12&quot; &quot;HOXC10&quot; &quot;HOXC11&quot; &quot;HOXB9&quot; &quot;ESRRA&quot; &quot;HOXA10&quot;
     [,17]  [,18]   [,19]   [,20]  [,21]  [,22]   [,23]   [,24]   [,25]  
[1,] &quot;RXRG&quot; &quot;VENTX&quot; &quot;HMGA1&quot; &quot;RXRA&quot; &quot;RORA&quot; &quot;ESRRB&quot; &quot;ESRRG&quot; &quot;NR5A2&quot; &quot;MYEF2&quot;
     [,26]   [,27] [,28]
[1,] &quot;NR5A1&quot; &quot;CDX&quot; &quot;IRF&quot;</code></pre>
<pre class="r"><code>for (i in LY75_TF_names){
  print(blood_gene_exp_norm$Description[startsWith(blood_gene_exp_norm$Description, as.character(i))] )
}</code></pre>
<pre><code>[1] &quot;SP110&quot;  &quot;SP140&quot;  &quot;SP140L&quot; &quot;SP100&quot;  &quot;SP1&quot;   
character(0)
[1] &quot;TFAP2A&quot;     &quot;TFAP2A-AS1&quot;
[1] &quot;HIF1AN&quot;    &quot;HIF1A-AS1&quot; &quot;HIF1A&quot;     &quot;HIF1A-AS2&quot; &quot;HIF1AP1&quot;  
[1] &quot;CREB1&quot;
[1] &quot;AHRR&quot; &quot;AHR&quot; 
[1] &quot;ARNT&quot;   &quot;ARNTL&quot;  &quot;ARNTL2&quot; &quot;ARNT2&quot; 
[1] &quot;ZNF75A&quot;
[1] &quot;TFAP2C&quot;
[1] &quot;ETV6&quot;
[1] &quot;HOXD12&quot;
[1] &quot;HOXC10&quot;
[1] &quot;HOXC11&quot;
[1] &quot;HOXB9&quot;
[1] &quot;ESRRA&quot;   &quot;ESRRAP1&quot; &quot;ESRRAP2&quot;
[1] &quot;HOXA10&quot;
[1] &quot;RXRG&quot;
[1] &quot;VENTXP7&quot; &quot;VENTXP4&quot; &quot;VENTXP5&quot; &quot;VENTXP6&quot; &quot;VENTX&quot;   &quot;VENTXP2&quot; &quot;VENTXP1&quot;
[1] &quot;HMGA1P8&quot; &quot;HMGA1P2&quot; &quot;HMGA1&quot;   &quot;HMGA1P7&quot; &quot;HMGA1P4&quot; &quot;HMGA1P5&quot; &quot;HMGA1P3&quot;
[8] &quot;HMGA1P6&quot; &quot;HMGA1P1&quot;
[1] &quot;RXRA&quot;
[1] &quot;RORA&quot;
[1] &quot;ESRRB&quot;
[1] &quot;ESRRG&quot;
[1] &quot;NR5A2&quot;
[1] &quot;MYEF2&quot;
[1] &quot;NR5A1&quot;
[1] &quot;CDX1&quot; &quot;CDX2&quot; &quot;CDX4&quot;
 [1] &quot;IRF6&quot;    &quot;IRF2BP2&quot; &quot;IRF2&quot;    &quot;IRF1&quot;    &quot;IRF4&quot;    &quot;IRF5&quot;    &quot;IRF5P1&quot; 
 [8] &quot;IRF7&quot;    &quot;IRF9&quot;    &quot;IRF2BPL&quot; &quot;IRF8&quot;    &quot;IRF2BP1&quot; &quot;IRF3&quot;   </code></pre>
<p>We leave aside 3 TF genes, <em>HF1H3B</em>, <em>CDX</em>, and <em>IRF</em>, which cannot be exactly found in GTEx.</p>
<p>We then extract expression data of <em>LY75</em> and the rest 25 TFs that are available in GTEx:</p>
<pre class="r"><code>LY75_TF_names &lt;- LY75_TF_names[-c(2,27,28)]
n = 25
LY75_TF_exp &lt;- data.frame(matrix(nrow = 407, ncol = n))

for (i in c(1:n)){
  LY75_TF_exp[,i] &lt;- t(blood_gene_exp_norm[which(blood_gene_exp_norm$Description
                                          ==as.character(LY75_TF_names[i])),-1])
}

LY75_exp &lt;- t(blood_gene_exp_norm[which(blood_gene_exp_norm$Description==&#39;LY75&#39;),-1])
LY75_TF_exp &lt;- cbind(LY75_TF_exp,LY75_exp)
colnames(LY75_TF_exp) &lt;- c(LY75_TF_names,&#39;LY75&#39;)
row.names(LY75_TF_exp) &lt;- colnames(blood_gene_exp_norm)[-1]</code></pre>
</div>
<div id="linear-regression" class="section level2">
<h2>Linear Regression</h2>
<p>Multiple linear regression on all 25 TFs:</p>
<pre class="r"><code>fit_all_LY75 &lt;- lm(LY75 ~ ., data = LY75_TF_exp)
summary(fit_all_LY75)</code></pre>
<pre><code>
Call:
lm(formula = LY75 ~ ., data = LY75_TF_exp)

Residuals:
       Min         1Q     Median         3Q        Max 
-8.368e-06 -1.008e-06 -1.252e-07  7.886e-07  9.383e-06 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -3.979e-07  3.700e-07  -1.076 0.282825    
SP1          1.093e-02  1.253e-02   0.872 0.383572    
TFAP2A       9.303e+00  4.225e+00   2.202 0.028285 *  
HIF1A        1.588e-02  2.514e-03   6.319 7.36e-10 ***
CREB1        1.665e-01  7.345e-02   2.266 0.023996 *  
AHR          1.144e-01  2.166e-02   5.283 2.14e-07 ***
ARNT        -4.017e-02  3.355e-02  -1.197 0.231957    
ZNF75A       2.434e-01  8.445e-02   2.882 0.004178 ** 
TFAP2C       1.334e+00  1.837e+00   0.726 0.468255    
ETV6        -8.604e-04  1.119e-02  -0.077 0.938764    
HOXD12      -3.735e+01  2.190e+01  -1.706 0.088856 .  
HOXC10       1.933e-01  3.704e-01   0.522 0.602053    
HOXC11       2.757e+00  3.402e+00   0.810 0.418299    
HOXB9       -1.245e+00  8.870e-01  -1.403 0.161332    
ESRRA        7.713e-03  1.734e-02   0.445 0.656740    
HOXA10      -2.933e-01  6.960e-01  -0.421 0.673716    
RXRG        -6.966e-01  6.026e-01  -1.156 0.248420    
VENTX       -4.720e-02  2.895e-02  -1.631 0.103808    
HMGA1        8.296e-03  3.900e-03   2.127 0.034036 *  
RXRA        -2.937e-03  1.140e-02  -0.258 0.796767    
RORA         1.446e-01  3.734e-02   3.872 0.000127 ***
ESRRB        2.673e+00  3.653e+00   0.732 0.464749    
ESRRG       -8.121e+00  1.247e+01  -0.651 0.515135    
NR5A2       -1.227e+00  3.121e+00  -0.393 0.694343    
MYEF2        9.293e-01  3.073e-01   3.024 0.002661 ** 
NR5A1        2.290e-01  5.782e-01   0.396 0.692252    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 2.237e-06 on 381 degrees of freedom
Multiple R-squared:  0.773, Adjusted R-squared:  0.7581 
F-statistic: 51.89 on 25 and 381 DF,  p-value: &lt; 2.2e-16</code></pre>
<p><strong>Forward selection</strong></p>
<p>Find the first significant predictor:</p>
<pre><code>[1] &quot;HIF1A&quot;</code></pre>
<p>Add rest of the predictors one at a time until none of their p-values are significant (p value &gt; 0.05):</p>
<p>The p-value and <span class="math inline">\(R^2\)</span> statistic for each of the significant predictors added at each round of selection are:</p>
<pre class="r"><code>stats_LY75</code></pre>
<pre><code>                 HIF1A       ZNF75A        MYEF2          AHR         RORA
p-val     7.079951e-67 7.276625e-45 7.074830e-07 8.710178e-09 0.0002697819
R-squared 5.218495e-01 7.069205e-01 7.242927e-01 7.461165e-01 0.7543903294
                CREB1     TFAP2A
p-val     0.006952932 0.01789507
R-squared 0.758828547 0.76219776</code></pre>
<p>Summary of the selected model:</p>
<pre class="r"><code>summary(forward_model_LY75)</code></pre>
<pre><code>
Call:
lm(formula = LY75 ~ HIF1A + ZNF75A + MYEF2 + AHR + RORA + CREB1 + 
    TFAP2A, data = df)

Residuals:
       Min         1Q     Median         3Q        Max 
-9.051e-06 -1.039e-06 -1.720e-07  8.557e-07  9.755e-06 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -3.237e-07  2.275e-07  -1.423 0.155465    
HIF1A        1.600e-02  1.761e-03   9.085  &lt; 2e-16 ***
ZNF75A       2.890e-01  7.233e-02   3.996 7.69e-05 ***
MYEF2        1.035e+00  2.768e-01   3.740 0.000211 ***
AHR          1.019e-01  1.692e-02   6.025 3.84e-09 ***
RORA         1.444e-01  3.045e-02   4.742 2.96e-06 ***
CREB1        1.233e-01  4.466e-02   2.760 0.006046 ** 
TFAP2A       8.982e+00  3.778e+00   2.378 0.017895 *  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 2.237e-06 on 399 degrees of freedom
Multiple R-squared:  0.7622,    Adjusted R-squared:  0.758 
F-statistic: 182.7 on 7 and 399 DF,  p-value: &lt; 2.2e-16</code></pre>
<p>Plot the increase of <span class="math inline">\(R^2\)</span> with the adding of predictors:</p>
<pre class="r"><code>plot(as.matrix(stats_LY75)[2,], pch=16, xlab=&quot;Number of Predictors&quot;,ylab = &quot;R_2&quot;)</code></pre>
<p><img src="Figs/unnamed-chunk-14-1.png" width="576" /></p>
<p><strong>Normality check and Boxcoc transformation</strong></p>
<p>Check the normality assumption:</p>
<pre class="r"><code>par(mfrow=c(1,2))
hist(fit_all_LY75$residuals)
qqnorm(fit_all_LY75$residuals,pch=16) 
qqline(fit_all_LY75$residuals,col=&#39;red&#39;)</code></pre>
<p><img src="Figs/unnamed-chunk-15-1.png" width="576" /></p>
<p>The distridution of residuals is heavily long-tailed.</p>
<p>We try to do Boxcox transformation on the response variable to restore the normality:</p>
<pre class="r"><code>library(MASS)
bc &lt;- boxcox(fit_all_LY75, plot=T, lambda=seq(-0.5,1,by=0.1))</code></pre>
<p><img src="Figs/unnamed-chunk-16-1.png" width="576" /></p>
<p><span class="math inline">\(\lambda\)</span> that maximizes the profile log-likelihood is:</p>
<pre class="r"><code>bc$x[which(bc$y == max(bc$y))]</code></pre>
<pre><code>[1] 0.3939394</code></pre>
<pre class="r"><code>lambda=0.4
LY75_TF_exp_bc &lt;- LY75_TF_exp
LY75_TF_exp_bc[,n+1] &lt;- (LY75_TF_exp[,n+1]^lambda-1)/lambda
fit_bc_LY75 &lt;- lm(LY75 ~., data=LY75_TF_exp_bc)
par(mfrow=c(1,2))
hist(fit_bc_LY75$residuals)
qqnorm(fit_bc_LY75$residuals,pch=16) 
qqline(fit_bc_LY75$residuals,col=&#39;red&#39;)</code></pre>
<p><img src="Figs/unnamed-chunk-18-1.png" width="576" /></p>
</div>
<div id="negative-controls" class="section level2">
<h2>Negative Controls</h2>
<p>We randomly select 7 genes available in GTEx and regress <em>LY75</em> expression onto their expression levels to see if <span class="math inline">\(R^2\)</span> is as big as our previous model in which the predictors are the 7 selected TFs.</p>
<pre class="r"><code>p = 7
r = 1000
R_sq_LY75 = rep(0,r)
LY75_TF_control &lt;- data.frame(matrix(nrow = 407, ncol = p))
LY75_TF_control &lt;- cbind(LY75_TF_control,t(LY75_exp))
colnames(LY75_TF_control)[p+1] &lt;- &#39;LY75&#39;

rand_LY75 = matrix(nrow = r, ncol = p)

for (k in c(1:r)){
  rand_LY75[k,] &lt;- sample(c(1:56202),p,replace = FALSE)
  rand_LY75_name &lt;- blood_gene_exp_norm$Description[rand_LY75[k,]]
  temp &lt;- rand_LY75_name[duplicated(rand_LY75_name)]

  while (length(temp)&gt;0){
    rand_LY75[k,] &lt;- sample(c(1:56202),p,replace = FALSE)
    rand_LY75_name &lt;- blood_gene_exp_norm$Description[rand_LY75[k,]]
    temp &lt;- rand_LY75_name[duplicated(rand_LY75_name)]
  }

  colnames(LY75_TF_control)[1:p] &lt;- rand_LY75_name

  for (i in c(1:p)){
    LY75_TF_control[,i] &lt;- t(blood_gene_exp_norm[rand_LY75[k,i],-1])
  }

  fit_control_LY75 &lt;- summary(lm(LY75 ~. , data = LY75_TF_control))
  R_sq_LY75[k] = fit_control_LY75$r.squared
}</code></pre>
<p><span class="math inline">\(R^2\)</span>s from the control models:</p>
<pre class="r"><code>hist(R_sq_LY75, main = &#39;Distribution of R_sq of Control Models&#39;)</code></pre>
<p><img src="Figs/unnamed-chunk-20-1.png" width="576" /></p>
<pre class="r"><code>summary(R_sq_LY75)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.0000  0.4999  0.5000  0.4990  0.5001  0.5002 </code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
