<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Yifan Zhou" />


<title>TFBS ChIP-seq Data Analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}

.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Gene Regulation</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Description</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Progress
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="enhancer_tss.html">Enhancer-TSS</a>
    </li>
    <li>
      <a href="prelim_TF_gene_analysis.html">TF-Gene(LY75)</a>
    </li>
    <li>
      <a href="HBB-TF_Regression_Analysis.html">TF-Gene(HBB)</a>
    </li>
    <li>
      <a href="LCL_analysis.html">LCL To-do List</a>
    </li>
    <li>
      <a href="TFBS_chipseq.html">LCL ChIP-seq TF-Gene Correlation</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">TFBS ChIP-seq Data Analysis</h1>
<h4 class="author"><em>Yifan Zhou</em></h4>
<h4 class="date"><em>2/20/2018</em></h4>

</div>


<div id="chip-seq-data" class="section level2">
<h2>ChIP-seq Data</h2>
<p>TF binding site ChIP-seq data in LCL are downloaded from the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?hgsid=657090723_G179QYBBXSSU1XvyOs35l9cq4BaC&amp;c=chrY&amp;g=wgEncodeAwgTfbsUniform">ENCODE Uniform TFBS track</a>:</p>
<ul>
<li>cell line: GM12878 (Tier 1)</li>
<li>file format: <a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format12">ENCODE narrowPeak</a></li>
</ul>
<p>Open chromatin regions in LCL:</p>
<p>DHS peaks are available from <a href="http://www.sciencedirect.com/science/article/pii/S0092867415009642?via%3Dihub#app1">Grubert et al., 2015</a> (Table S2), which were summarized over the original DHS profiles of 70 different LCL cell lines in <a href="https://www.nature.com/articles/nature10808#accessions">Degner et al., 2012</a>.</p>
<ul>
<li>~ 250,000 DHS peaks in total</li>
</ul>
<div id="tfs-of-interest" class="section level3">
<h3>TFs of interest:</h3>
<p><strong>YY1</strong></p>
<p>Encode ChIP-seq file name: <code>wgEncodeAwgTfbsSydhGm12878Yy1UniPk.narrowPeak</code>, integrated quality flag: Caution.</p>
<p><strong>STAT1</strong></p>
<p>Encode ChIP-seq file name: <code>wgEncodeAwgTfbsSydhGm12878Stat1UniPk.narrowPeak</code>, integrated quality flag: Caution.</p>
<p><strong>CEBPB</strong></p>
<p>Encode ChIP-seq file name: <code>wgEncodeAwgTfbsHaibGm12878Cebpbsc150V0422111UniPk.narrowPeak</code>, integrated quality flag: Caution.</p>
<p><strong>SP1</strong></p>
<p>Encode ChIP-seq file name: <code>wgEncodeAwgTfbsHaibGm12878Sp1Pcr1xUniPk.narrowPeak</code>, integrated quality flag: Good.</p>
<p><strong>NF-YA</strong></p>
<p>File name: <code>wgEncodeAwgTfbsSydhGm12878NfyaIggmusUniPk.narrowPeak</code>, integrated quality flag: Good.</p>
</div>
<div id="data-summary" class="section level3">
<h3>Data Summary:</h3>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">YY1</th>
<th align="right">STAT1</th>
<th align="right">CEBPB</th>
<th align="right">SP1</th>
<th align="right">NFYA</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td># of ChIP-seq peaks</td>
<td align="right">2077</td>
<td align="right">1769</td>
<td align="right">5786</td>
<td align="right">18248</td>
<td align="right">1841</td>
</tr>
<tr class="even">
<td>% of peaks in DHS</td>
<td align="right">67</td>
<td align="right">83</td>
<td align="right">92</td>
<td align="right">95</td>
<td align="right">96</td>
</tr>
<tr class="odd">
<td># of DHS overlapped</td>
<td align="right">1702</td>
<td align="right">1970</td>
<td align="right">6685</td>
<td align="right">18543</td>
<td align="right">1897</td>
</tr>
<tr class="even">
<td># of genes in contact w/ those DHS</td>
<td align="right">1184</td>
<td align="right">1603</td>
<td align="right">2577</td>
<td align="right">9282</td>
<td align="right">2273</td>
</tr>
<tr class="odd">
<td>% of peaks in 5kb gene promoters</td>
<td align="right">26</td>
<td align="right">36</td>
<td align="right">15</td>
<td align="right">36</td>
<td align="right">64</td>
</tr>
<tr class="even">
<td># of 5kb gene promoters overlapped</td>
<td align="right">617</td>
<td align="right">727</td>
<td align="right">986</td>
<td align="right">6722</td>
<td align="right">1397</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="analysis-of-expression-correlation" class="section level2">
<h2>Analysis of Expression Correlation</h2>
<p>Analyze the correlation of mRNA expression levels btw a certain TF and all the other genes (one at a time), compare btw target genes &amp; non-target genes:</p>
<ul>
<li>RNA-seq data: 16614 genes, 75 LCL cell lines (<a href="http://science.sciencemag.org/content/suppl/2014/12/17/science.1260793.DC1?_ga=2.48064679.1826706411.1515443144-1380778862.1504648991">Battle et al., 2015</a>, Data Table S4).</li>
</ul>
<ul>
<li>Target genes are defined as those of which the +5kb promoter regions overlap with the TF ChIP-seq peaks.</li>
</ul>
<div id="tf-of-interest-stat1" class="section level3">
<h3>TF of interest: STAT1</h3>
<p>[<em>STAT1</em>] ~ [<em>gene</em>] mRNA expression level correlation (<span class="math inline">\(R^2\)</span>) distributions:</p>
<ul>
<li>Target genes linked through ChIP-seq peaks with H3K27ac &amp; in 3kb upstream gene promoter regions:</li>
</ul>
<pre><code>[1] &quot;Number of target genes: 671&quot;
[1] &quot;Proportion among all genes: 0.0404&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-4-1.png" width="768" /></p>
<p><img src="Figs/unnamed-chunk-5-1.png" width="768" /></p>
<ul>
<li>After correcting for the top 2 PCs:</li>
</ul>
<pre><code>[1] &quot;Number of target genes: 671&quot;
[1] &quot;Proportion among all genes: 0.0404&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-7-1.png" width="768" /></p>
<ul>
<li>Further pick the top 10% peaks based on their ChIP-se enrichment signal values, and link the target genes similarly through 3kb promoters (also corrected for the top 2 PCs):</li>
</ul>
<pre><code>[1] &quot;Number of target genes: 109&quot;
[1] &quot;Proportion among all genes: 0.0066&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-9-1.png" width="768" /></p>
<ul>
<li>Target genes linked through ChIP-seq peaks in DHS+H3K27ac regions that are in Hi-C contact with gene promoters (also corrected for the top 2 PCs):</li>
</ul>
<pre><code>[1] &quot;Number of target genes: 1534&quot;
[1] &quot;Proportion among all genes: 0.0923&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-11-1.png" width="768" /></p>
<ul>
<li>Further pick the top 10% peaks based on their ChIP-se enrichment signal values, and link the target genes similarly through Hi-C (also corrected for the top 2 PCs):</li>
</ul>
<pre><code>[1] &quot;Number of target genes: 218&quot;
[1] &quot;Proportion among all genes: 0.0131&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-13-1.png" width="768" /></p>
<ul>
<li>Peaks that contain STAT1 motif seqeunces from JASPAR:</li>
</ul>
<pre><code>[1] &quot;Number of target genes: 163&quot;
[1] &quot;Proportion among all genes: 0.0098&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-14-1.png" width="768" /></p>
<ul>
<li>Peaks that overlap with strong enhancers from ChromHmm, target genes linked through HiC:</li>
</ul>
<pre><code>[1] &quot;Number of target genes: 366&quot;
[1] &quot;Proportion among all genes: 0.022&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-16-1.png" width="768" /></p>
</div>
<div id="tf-of-interest-yy1" class="section level3">
<h3>TF of interest: YY1</h3>
<p>[<em>YY1</em>] ~ [<em>gene</em>] mRNA expression level correlation (<span class="math inline">\(R^2\)</span>) distributions:</p>
<pre><code>[1] &quot;Number of target genes: 567&quot;
[1] &quot;Proportion among all genes: 0.0341&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-17-1.png" width="768" /></p>
<pre><code>[1] &quot;Number of target genes: 567&quot;
[1] &quot;Proportion among all genes: 0.0341&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-18-1.png" width="768" /></p>
<pre><code>[1] &quot;Number of target genes: 364&quot;
[1] &quot;Proportion among all genes: 0.0219&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-19-1.png" width="768" /></p>
</div>
<div id="tf-of-interest-cebpb" class="section level3">
<h3>TF of interest: CEBPB</h3>
<p>[<em>CEBPB</em>] ~ [<em>gene</em>] mRNA expression level correlation (<span class="math inline">\(R^2\)</span>) distributions:</p>
<pre><code>[1] &quot;Number of target genes: 852&quot;
[1] &quot;Proportion among all genes: 0.0513&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-20-1.png" width="768" /></p>
<pre><code>[1] &quot;Number of target genes: 852&quot;
[1] &quot;Proportion among all genes: 0.0513&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-21-1.png" width="768" /></p>
</div>
<div id="tf-of-interest-sp1" class="section level3">
<h3>TF of interest: SP1</h3>
<p>[<em>Sp1</em>] ~ [<em>gene</em>] mRNA expression level correlation (<span class="math inline">\(R^2\)</span>) distributions:</p>
<pre><code>[1] &quot;Number of target genes: 6441&quot;
[1] &quot;Proportion among all genes: 0.3877&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-22-1.png" width="768" /></p>
<pre><code>[1] &quot;Number of target genes: 6441&quot;
[1] &quot;Proportion among all genes: 0.3877&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-23-1.png" width="768" /></p>
<pre><code>[1] &quot;Number of target genes: 1965&quot;
[1] &quot;Proportion among all genes: 0.1183&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-24-1.png" width="768" /></p>
<pre><code>[1] &quot;Number of target genes: 3015&quot;
[1] &quot;Proportion among all genes: 0.1815&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-25-1.png" width="768" /></p>
<pre><code>[1] &quot;Number of target genes: 3466&quot;
[1] &quot;Proportion among all genes: 0.2086&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-26-1.png" width="768" /></p>
</div>
<div id="tf-of-interest-nf-ya" class="section level3">
<h3>TF of interest: NF-YA</h3>
<p>[<em>NF-YA</em>] ~ [<em>gene</em>] mRNA expression level correlation (<span class="math inline">\(R^2\)</span>) distributions:</p>
<pre><code>[1] &quot;Number of target genes: 1331&quot;
[1] &quot;Proportion among all genes: 0.0801&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-27-1.png" width="768" /></p>
<pre><code>[1] &quot;Number of target genes: 1331&quot;
[1] &quot;Proportion among all genes: 0.0801&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-28-1.png" width="768" /></p>
<pre><code>[1] &quot;Number of target genes: 206&quot;
[1] &quot;Proportion among all genes: 0.0124&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-29-1.png" width="768" /></p>
</div>
<div id="tf-of-interest-irf4" class="section level3">
<h3>TF of interest: IRF4</h3>
<p>[<em>IRF4</em>] ~ [<em>gene</em>] mRNA expression level correlation (<span class="math inline">\(R^2\)</span>) distributions:</p>
<pre><code>[1] &quot;Number of target genes: 2130&quot;
[1] &quot;Proportion among all genes: 0.1282&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-30-1.png" width="768" /></p>
<pre><code>[1] &quot;Number of target genes: 2130&quot;
[1] &quot;Proportion among all genes: 0.1282&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-31-1.png" width="768" /></p>
<pre><code>[1] &quot;Number of target genes: 74&quot;
[1] &quot;Proportion among all genes: 0.0045&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-32-1.png" width="768" /></p>
<pre><code>[1] &quot;Number of target genes: 242&quot;
[1] &quot;Proportion among all genes: 0.0146&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-33-1.png" width="768" /></p>
<pre><code>[1] &quot;Number of target genes: 3259&quot;
[1] &quot;Proportion among all genes: 0.1962&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-34-1.png" width="768" /></p>
</div>
</div>
<div id="new-mrna-expression-data-geuvadis-database" class="section level2">
<h2>New mRNA expression data: Geuvadis database</h2>
<ul>
<li>RNA-seq gene expression data: <a href="https://www.ebi.ac.uk/arrayexpress/files/E-GEUV-1/analysis_results/">GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.gz</a></li>
<li>23722 genes, 462 LCL cell lines from the 1000 Genomes</li>
<li>Peer-factor normalized RPKM: library depth &amp; transcript lengths &amp; expressed &amp; PEER</li>
<li>14102 genes overlap with Battle dataset</li>
</ul>
<div id="r2-difference-between-2-datasets" class="section level3">
<h3><span class="math inline">\(R^2\)</span> difference between 2 datasets</h3>
<p>We focus on the 14102 genes that are shared between both the Battle dataset and the Geudavis dataset.<br />
For each TF, we calculate the correlation (<span class="math inline">\(R^2\)</span>) of its expression level with each one of the 14102 genes, using the Geudavis dataset first and the Battle dataset next. We plot the <span class="math inline">\(R^2\)</span>s obtained from the first dataset in increasing order (red), and then plot the <span class="math inline">\(R^2\)</span>s obtained from the first dataset for the corresponding genes (light blue), as we expect to see some level of correlation between these two groups of <span class="math inline">\(R^2\)</span>s. We then further calculated the correlation coefficient of these two groups of <span class="math inline">\(R^2\)</span>s.</p>
<p><img src="Figs/unnamed-chunk-35-1.png" width="768" /></p>
<pre><code>[1] &quot;Correlation btw the 2 groups of R^2&#39;s: 0.0264&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-36-1.png" width="768" /></p>
<pre><code>[1] &quot;Correlation btw the 2 groups of R^2&#39;s: 0.002&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-37-1.png" width="768" /></p>
<pre><code>[1] &quot;Correlation btw the 2 groups of R^2&#39;s: 0.0483&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-38-1.png" width="768" /></p>
<pre><code>[1] &quot;Correlation btw the 2 groups of R^2&#39;s: 0.0119&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-39-1.png" width="768" /></p>
<pre><code>[1] &quot;Correlation btw the 2 groups of R^2&#39;s: 0.0165&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-40-1.png" width="768" /></p>
<pre><code>[1] &quot;Correlation btw the 2 groups of R^2&#39;s: 0.0075&quot;</code></pre>
</div>
<div id="strong-enhancers" class="section level3">
<h3>Strong Enhancers</h3>
<p>We use the annotations from the ChromHmm file <a href="http://rohsdb.cmb.usc.edu/GBshape/cgi-bin/hgTrackUi?db=hg19&amp;g=wgEncodeBroadHmm">wgEncodeBroadHmmGm12878HMM.bed.gz</a> to define the “strong enhancer” regions on the human reference genome (<em>hg19</em>) in LCL cell line <em>GM12787</em>.</p>
</div>
<div id="correlation-analysis" class="section level3">
<h3>Correlation Analysis</h3>
<ul>
<li>STAT1, target genes: ChIP-seq peaks in 3kb promoter regions</li>
</ul>
<pre><code>[1] &quot;Number of target genes: 934&quot;
[1] &quot;Proportion among all genes: 0.0394&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-41-1.png" width="768" /></p>
<ul>
<li>STAT1, target genes: ChIP-seq peaks in ChromHmm “strong enhancers”in Hi-C contact with promoter regions</li>
</ul>
<pre><code>[1] &quot;Number of target genes: 503&quot;
[1] &quot;Proportion among all genes: 0.0212&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-42-1.png" width="768" /></p>
<ul>
<li>YY1, target genes: ChIP-seq peaks in 3kb promoter regions</li>
</ul>
<pre><code>[1] &quot;Number of target genes: 742&quot;
[1] &quot;Proportion among all genes: 0.0313&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-43-1.png" width="768" /></p>
<ul>
<li>YY1, target genes: ChIP-seq peaks in ChromHmm “strong enhancers” that are in contact with promoter regions through Hi-C</li>
</ul>
<pre><code>[1] &quot;Number of target genes: 496&quot;
[1] &quot;Proportion among all genes: 0.0209&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-44-1.png" width="768" /></p>
<ul>
<li>CEBPB, target genes: ChIP-seq peaks in 3kb promoter regions</li>
</ul>
<pre><code>[1] &quot;Number of target genes: 1085&quot;</code></pre>
<pre><code>[1] &quot;Proportion among all genes: 0.0457&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-45-1.png" width="768" /></p>
<ul>
<li>CEBPB, target genes: ChIP-seq peaks in ChromHmm “strong enhancers” that are in contact with promoter regions through Hi-C</li>
</ul>
<pre><code>[1] &quot;Number of target genes: 2515&quot;</code></pre>
<pre><code>[1] &quot;Proportion among all genes: 0.106&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-46-1.png" width="768" /></p>
<ul>
<li>SP1, target genes: ChIP-seq peaks in 3kb promoter regions</li>
</ul>
<pre><code>[1] &quot;Number of target genes: 7705&quot;
[1] &quot;Proportion among all genes: 0.3248&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-47-1.png" width="768" /></p>
<ul>
<li>SP1, target genes: ChIP-seq peaks in ChromHmm “strong enhancers” that are in contact with promoter regions through Hi-C</li>
</ul>
<pre><code>[1] &quot;Number of target genes: 4798&quot;
[1] &quot;Proportion among all genes: 0.2023&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-48-1.png" width="768" /></p>
<ul>
<li>NFYA, target genes: ChIP-seq peaks in 3kb promoter regions</li>
</ul>
<pre><code>[1] &quot;Number of target genes: 1545&quot;</code></pre>
<pre><code>[1] &quot;Proportion among all genes: 0.0651&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-49-1.png" width="768" /></p>
<ul>
<li>NFYA, target genes: ChIP-seq peaks in ChromHmm “strong enhancers” that are in contact with promoter regions through Hi-C</li>
</ul>
<pre><code>[1] &quot;Number of target genes: 279&quot;
[1] &quot;Proportion among all genes: 0.0118&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-50-1.png" width="768" /></p>
<ul>
<li>IRF4, target genes: ChIP-seq peaks in 3kb promoter regions</li>
</ul>
<pre><code>[1] &quot;Number of target genes: 2709&quot;
[1] &quot;Proportion among all genes: 0.1142&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-51-1.png" width="768" /></p>
<ul>
<li>IRF4, target genes: ChIP-seq peaks in ChromHmm “strong enhancers” that are in contact with promoter regions through Hi-C</li>
</ul>
<pre><code>[1] &quot;Number of target genes: 4647&quot;
[1] &quot;Proportion among all genes: 0.1959&quot;</code></pre>
<p><img src="Figs/unnamed-chunk-52-1.png" width="768" /></p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
